{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up and install packages if they are missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for installing packages\n",
    "install_if_missing <- function(packages) {\n",
    "    if (length(setdiff(packages, rownames(installed.packages()))) > 0) {\n",
    "        install.packages(setdiff(packages, rownames(installed.packages())))\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "install_if_missing(c('tidyverse', 'viridis', 'ggthemes', 'pryr', 'skimr', 'testthat', 'reticulate', 'WebGestaltR', 'readxl'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(viridis)    # A nice color scheme for plots.\n",
    "library(ggthemes)   # Common themes to change the look and feel of plots.\n",
    "library(scales)     # Graphical scales map data to aesthetics in plots.\n",
    "library(testthat)   # Testing functions.\n",
    "library(assertthat) # Assertion functions.\n",
    "library(pryr)       # Memory usage functions.\n",
    "library(skimr)      # Summary statistics for dataframes.\n",
    "library(bigrquery)  # BigQuery R client.\n",
    "library(tidyverse)  # Data wrangling packages.\n",
    "library(reticulate)  # R Interface to Python\n",
    "\n",
    "library(Ronaldo)    # Leonardo R package.\n",
    "\n",
    "library(data.table) # Data Table package for faster reading and processing\n",
    "\n",
    "\n",
    "library(reticulate) # reticulate for calling the FireCloud Python API\n",
    "\n",
    "library(ggplot2)    # Load ggplot2 for graphs\n",
    "\n",
    "library(bigrquery)  # Load biqrquery for interacting with BigQuery\n",
    "\n",
    "library(WebGestaltR)# For pathway/gene set enrichment analysis for this pipeline\n",
    "\n",
    "library(readxl)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up environment variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "BILLING_PROJECT_ID <- Sys.getenv('GOOGLE_PROJECT')\n",
    "WORKSPACE_NAMESPACE <- Sys.getenv('WORKSPACE_NAMESPACE')\n",
    "WORKSPACE_NAME <- Sys.getenv('WORKSPACE_NAME')\n",
    "WORKSPACE_BUCKET <-  Sys.getenv('WORKSPACE_BUCKET')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up utility functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "shell_do <- function(command) {\n",
    "    print(paste('Executing: ', command))\n",
    "    system(command, intern = TRUE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routines for reading files from Google Cloud Storage\n",
    "gcs_read_file <- function(path) {\n",
    "    pipe(str_glue('gsutil -u {BILLING_PROJECT_ID} cat {path}'))\n",
    "}\n",
    "gcs_read_csv <- function(path, sep=',') {\n",
    "    readr::read_csv(gcs_read_file(path))\n",
    "}\n",
    "\n",
    "# Utility routines for reading files from Google BigQuery\n",
    "bq_query <- function(query) {\n",
    "    # Return the contents of a query against BigQuery    \n",
    "    return(bigrquery::bq_table_download(\n",
    "        bigrquery::bq_project_query(BILLING_PROJECT_ID, query = query)))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import GRCh37 gene coordinates into workspace"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NCBI genome build 37.3 (this is an old version)\n",
    "\n",
    "Obtained from MAGMA auxilary files: https://ctg.cncr.nl/software/magma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Executing:  gsutil -mu gp2-ipdgc-hackathon cp gs://fc-secure-b9f9d17f-b38c-407d-85e0-7a759f13cea0/NCBI37.3.zip ~/bin/data_temp/\"\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shell_do(str_glue('gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/NCBI37.3.zip ~/bin/data_temp/'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'nallsEtAl2019_excluding23andMe_allVariants.tab'</li><li>'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'</li><li>'NCBI37.3.zip'</li><li>'PDGWAS_for_FUMA_MT.txt'</li><li>'PDGWAS_for_FUMA_MT.txt.zip'</li><li>'test.zip'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab'\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab.zip'\n",
       "\\item 'NCBI37.3.zip'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt.zip'\n",
       "\\item 'test.zip'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'nallsEtAl2019_excluding23andMe_allVariants.tab'\n",
       "2. 'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'\n",
       "3. 'NCBI37.3.zip'\n",
       "4. 'PDGWAS_for_FUMA_MT.txt'\n",
       "5. 'PDGWAS_for_FUMA_MT.txt.zip'\n",
       "6. 'test.zip'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"nallsEtAl2019_excluding23andMe_allVariants.tab\"    \n",
       "[2] \"nallsEtAl2019_excluding23andMe_allVariants.tab.zip\"\n",
       "[3] \"NCBI37.3.zip\"                                      \n",
       "[4] \"PDGWAS_for_FUMA_MT.txt\"                            \n",
       "[5] \"PDGWAS_for_FUMA_MT.txt.zip\"                        \n",
       "[6] \"test.zip\"                                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#List files in ~/bin/data_temp/\n",
    "system('ls ~/bin/data_temp/', intern=TRUE) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in system(\"unzip ~/bin/data_temp/NCBI37.3.zip\", intern = TRUE):\n",
      "“running command 'unzip ~/bin/data_temp/NCBI37.3.zip' had status 1”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span style=white-space:pre-wrap>'Archive:  /home/jupyter-user/bin/data_temp/NCBI37.3.zip'</span>"
      ],
      "text/latex": [
       "'Archive:  /home/jupyter-user/bin/data\\_temp/NCBI37.3.zip'"
      ],
      "text/markdown": [
       "<span style=white-space:pre-wrap>'Archive:  /home/jupyter-user/bin/data_temp/NCBI37.3.zip'</span>"
      ],
      "text/plain": [
       "[1] \"Archive:  /home/jupyter-user/bin/data_temp/NCBI37.3.zip\"\n",
       "attr(,\"status\")\n",
       "[1] 1\n",
       "attr(,\"errmsg\")\n",
       "[1] \"Resource temporarily unavailable\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Unzip gene coordinates file\n",
    "system('unzip ~/bin/data_temp/NCBI37.3.zip', intern=TRUE) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download current genome build gene coordinates\n",
    "\n",
    "Downloaded from: http://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "system('wget http://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Anaconda3-2020.11-Linux-x86_64.sh'</li><li>'format_for_FUMA_Manuela.ipynb'</li><li>'FUMA_format_input_file_Amica.ipynb'</li><li>'Homo_sapiens.GRCh37.87.gtf'</li><li>'import_GWAS_sumstats_Manuela.ipynb'</li><li>'ldsc'</li><li>'Miniconda3-latest-Linux-x86_64.sh'</li><li>'Miniconda3-latest-Linux-x86.sh'</li><li>'NCBI37.3.gene.loc'</li><li>'proj.ipynb'</li><li>'Python_Manuela.ipynb'</li><li>'README'</li><li>'REPORT'</li><li>'run_WebGestaltR_Manuela.ipynb'</li><li>'stratified_LDSC_Manuela.ipynb'</li><li>'WebGestaltR_VI.ipynb'</li><li>'{WORKSPACE_BUCKET}'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Anaconda3-2020.11-Linux-x86\\_64.sh'\n",
       "\\item 'format\\_for\\_FUMA\\_Manuela.ipynb'\n",
       "\\item 'FUMA\\_format\\_input\\_file\\_Amica.ipynb'\n",
       "\\item 'Homo\\_sapiens.GRCh37.87.gtf'\n",
       "\\item 'import\\_GWAS\\_sumstats\\_Manuela.ipynb'\n",
       "\\item 'ldsc'\n",
       "\\item 'Miniconda3-latest-Linux-x86\\_64.sh'\n",
       "\\item 'Miniconda3-latest-Linux-x86.sh'\n",
       "\\item 'NCBI37.3.gene.loc'\n",
       "\\item 'proj.ipynb'\n",
       "\\item 'Python\\_Manuela.ipynb'\n",
       "\\item 'README'\n",
       "\\item 'REPORT'\n",
       "\\item 'run\\_WebGestaltR\\_Manuela.ipynb'\n",
       "\\item 'stratified\\_LDSC\\_Manuela.ipynb'\n",
       "\\item 'WebGestaltR\\_VI.ipynb'\n",
       "\\item '\\{WORKSPACE\\_BUCKET\\}'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Anaconda3-2020.11-Linux-x86_64.sh'\n",
       "2. 'format_for_FUMA_Manuela.ipynb'\n",
       "3. 'FUMA_format_input_file_Amica.ipynb'\n",
       "4. 'Homo_sapiens.GRCh37.87.gtf'\n",
       "5. 'import_GWAS_sumstats_Manuela.ipynb'\n",
       "6. 'ldsc'\n",
       "7. 'Miniconda3-latest-Linux-x86_64.sh'\n",
       "8. 'Miniconda3-latest-Linux-x86.sh'\n",
       "9. 'NCBI37.3.gene.loc'\n",
       "10. 'proj.ipynb'\n",
       "11. 'Python_Manuela.ipynb'\n",
       "12. 'README'\n",
       "13. 'REPORT'\n",
       "14. 'run_WebGestaltR_Manuela.ipynb'\n",
       "15. 'stratified_LDSC_Manuela.ipynb'\n",
       "16. 'WebGestaltR_VI.ipynb'\n",
       "17. '{WORKSPACE_BUCKET}'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Anaconda3-2020.11-Linux-x86_64.sh\"  \"format_for_FUMA_Manuela.ipynb\"     \n",
       " [3] \"FUMA_format_input_file_Amica.ipynb\" \"Homo_sapiens.GRCh37.87.gtf\"        \n",
       " [5] \"import_GWAS_sumstats_Manuela.ipynb\" \"ldsc\"                              \n",
       " [7] \"Miniconda3-latest-Linux-x86_64.sh\"  \"Miniconda3-latest-Linux-x86.sh\"    \n",
       " [9] \"NCBI37.3.gene.loc\"                  \"proj.ipynb\"                        \n",
       "[11] \"Python_Manuela.ipynb\"               \"README\"                            \n",
       "[13] \"REPORT\"                             \"run_WebGestaltR_Manuela.ipynb\"     \n",
       "[15] \"stratified_LDSC_Manuela.ipynb\"      \"WebGestaltR_VI.ipynb\"              \n",
       "[17] \"{WORKSPACE_BUCKET}\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('ls', intern = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('gunzip Homo_sapiens.GRCh37.87.gtf.gz', intern=TRUE) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in PD GWAS summary statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "data <- fread(\"~/bin/data_temp/nallsEtAl2019_excluding23andMe_allVariants.tab\", header = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>freq</th><th scope=col>b</th><th scope=col>se</th><th scope=col>p</th><th scope=col>N_cases</th><th scope=col>N_controls</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr11:88249377</td><td>T</td><td>C</td><td>0.9931</td><td> 0.1575</td><td>0.1783</td><td>0.3771</td><td> 7161</td><td>  5356</td></tr>\n",
       "\t<tr><td>chr1:60320992 </td><td>A</td><td>G</td><td>0.9336</td><td> 0.0605</td><td>0.0456</td><td>0.1846</td><td>26421</td><td>442271</td></tr>\n",
       "\t<tr><td>chr2:18069070 </td><td>T</td><td>C</td><td>0.9988</td><td>-0.6774</td><td>1.3519</td><td>0.6163</td><td>  582</td><td>   905</td></tr>\n",
       "\t<tr><td>chr8:135908647</td><td>A</td><td>G</td><td>0.2081</td><td>-0.0358</td><td>0.0273</td><td>0.1887</td><td>26421</td><td>442271</td></tr>\n",
       "\t<tr><td>chr12:3871714 </td><td>A</td><td>C</td><td>0.9972</td><td> 0.1489</td><td>1.0636</td><td>0.8886</td><td>  749</td><td>   658</td></tr>\n",
       "\t<tr><td>chr16:77148858</td><td>A</td><td>G</td><td>0.9976</td><td>-0.1213</td><td>0.3874</td><td>0.7543</td><td> 6248</td><td>  4391</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " SNP & A1 & A2 & freq & b & se & p & N\\_cases & N\\_controls\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr11:88249377 & T & C & 0.9931 &  0.1575 & 0.1783 & 0.3771 &  7161 &   5356\\\\\n",
       "\t chr1:60320992  & A & G & 0.9336 &  0.0605 & 0.0456 & 0.1846 & 26421 & 442271\\\\\n",
       "\t chr2:18069070  & T & C & 0.9988 & -0.6774 & 1.3519 & 0.6163 &   582 &    905\\\\\n",
       "\t chr8:135908647 & A & G & 0.2081 & -0.0358 & 0.0273 & 0.1887 & 26421 & 442271\\\\\n",
       "\t chr12:3871714  & A & C & 0.9972 &  0.1489 & 1.0636 & 0.8886 &   749 &    658\\\\\n",
       "\t chr16:77148858 & A & G & 0.9976 & -0.1213 & 0.3874 & 0.7543 &  6248 &   4391\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 9\n",
       "\n",
       "| SNP &lt;chr&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | freq &lt;dbl&gt; | b &lt;dbl&gt; | se &lt;dbl&gt; | p &lt;dbl&gt; | N_cases &lt;int&gt; | N_controls &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| chr11:88249377 | T | C | 0.9931 |  0.1575 | 0.1783 | 0.3771 |  7161 |   5356 |\n",
       "| chr1:60320992  | A | G | 0.9336 |  0.0605 | 0.0456 | 0.1846 | 26421 | 442271 |\n",
       "| chr2:18069070  | T | C | 0.9988 | -0.6774 | 1.3519 | 0.6163 |   582 |    905 |\n",
       "| chr8:135908647 | A | G | 0.2081 | -0.0358 | 0.0273 | 0.1887 | 26421 | 442271 |\n",
       "| chr12:3871714  | A | C | 0.9972 |  0.1489 | 1.0636 | 0.8886 |   749 |    658 |\n",
       "| chr16:77148858 | A | G | 0.9976 | -0.1213 | 0.3874 | 0.7543 |  6248 |   4391 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP            A1 A2 freq   b       se     p      N_cases N_controls\n",
       "1 chr11:88249377 T  C  0.9931  0.1575 0.1783 0.3771  7161     5356    \n",
       "2 chr1:60320992  A  G  0.9336  0.0605 0.0456 0.1846 26421   442271    \n",
       "3 chr2:18069070  T  C  0.9988 -0.6774 1.3519 0.6163   582      905    \n",
       "4 chr8:135908647 A  G  0.2081 -0.0358 0.0273 0.1887 26421   442271    \n",
       "5 chr12:3871714  A  C  0.9972  0.1489 1.0636 0.8886   749      658    \n",
       "6 chr16:77148858 A  G  0.9976 -0.1213 0.3874 0.7543  6248     4391    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter for SNPs with p-value < 5e-08"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_filtered <- data %>%\n",
    "    filter(p < 5e-08)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>3464</li><li>9</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 3464\n",
       "\\item 9\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 3464\n",
       "2. 9\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 3464    9"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(data_filtered)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in gene coordinates file and tidy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_coords <- fread(\"Homo_sapiens.GRCh37.87.gtf\", skip = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene      </td><td>11869</td><td>14412</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>transcript</td><td>11869</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; tag \"basic\";                                                              </td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>exon      </td><td>11869</td><td>12227</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"1\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002234944\"; exon_version \"1\"; tag \"basic\";</td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>exon      </td><td>12613</td><td>12721</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00003582793\"; exon_version \"1\"; tag \"basic\";</td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>exon      </td><td>13221</td><td>14409</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"3\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002312635\"; exon_version \"1\"; tag \"basic\";</td></tr>\n",
       "\t<tr><td>1</td><td>ensembl       </td><td>transcript</td><td>11872</td><td>14412</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000515242\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-201\"; transcript_source \"ensembl\"; transcript_biotype \"transcribed_unprocessed_pseudogene\";                                                                                                                                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & ensembl\\_havana & gene       & 11869 & 14412 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\";                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\t 1 & havana         & transcript & 11869 & 14409 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; transcript\\_id \"ENST00000456328\"; transcript\\_version \"2\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; transcript\\_name \"DDX11L1-002\"; transcript\\_source \"havana\"; transcript\\_biotype \"processed\\_transcript\"; havana\\_transcript \"OTTHUMT00000362751\"; havana\\_transcript\\_version \"1\"; tag \"basic\";                                                              \\\\\n",
       "\t 1 & havana         & exon       & 11869 & 12227 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; transcript\\_id \"ENST00000456328\"; transcript\\_version \"2\"; exon\\_number \"1\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; transcript\\_name \"DDX11L1-002\"; transcript\\_source \"havana\"; transcript\\_biotype \"processed\\_transcript\"; havana\\_transcript \"OTTHUMT00000362751\"; havana\\_transcript\\_version \"1\"; exon\\_id \"ENSE00002234944\"; exon\\_version \"1\"; tag \"basic\";\\\\\n",
       "\t 1 & havana         & exon       & 12613 & 12721 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; transcript\\_id \"ENST00000456328\"; transcript\\_version \"2\"; exon\\_number \"2\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; transcript\\_name \"DDX11L1-002\"; transcript\\_source \"havana\"; transcript\\_biotype \"processed\\_transcript\"; havana\\_transcript \"OTTHUMT00000362751\"; havana\\_transcript\\_version \"1\"; exon\\_id \"ENSE00003582793\"; exon\\_version \"1\"; tag \"basic\";\\\\\n",
       "\t 1 & havana         & exon       & 13221 & 14409 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; transcript\\_id \"ENST00000456328\"; transcript\\_version \"2\"; exon\\_number \"3\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; transcript\\_name \"DDX11L1-002\"; transcript\\_source \"havana\"; transcript\\_biotype \"processed\\_transcript\"; havana\\_transcript \"OTTHUMT00000362751\"; havana\\_transcript\\_version \"1\"; exon\\_id \"ENSE00002312635\"; exon\\_version \"1\"; tag \"basic\";\\\\\n",
       "\t 1 & ensembl        & transcript & 11872 & 14412 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; transcript\\_id \"ENST00000515242\"; transcript\\_version \"2\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; transcript\\_name \"DDX11L1-201\"; transcript\\_source \"ensembl\"; transcript\\_biotype \"transcribed\\_unprocessed\\_pseudogene\";                                                                                                                                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 9\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;chr&gt; | V3 &lt;chr&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ensembl_havana | gene       | 11869 | 14412 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";                                                                                                                                                                                                                                                                                                                  |\n",
       "| 1 | havana         | transcript | 11869 | 14409 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; tag \"basic\";                                                               |\n",
       "| 1 | havana         | exon       | 11869 | 12227 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"1\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002234944\"; exon_version \"1\"; tag \"basic\"; |\n",
       "| 1 | havana         | exon       | 12613 | 12721 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00003582793\"; exon_version \"1\"; tag \"basic\"; |\n",
       "| 1 | havana         | exon       | 13221 | 14409 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"3\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002312635\"; exon_version \"1\"; tag \"basic\"; |\n",
       "| 1 | ensembl        | transcript | 11872 | 14412 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000515242\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-201\"; transcript_source \"ensembl\"; transcript_biotype \"transcribed_unprocessed_pseudogene\";                                                                                                                                    |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2             V3         V4    V5    V6 V7 V8\n",
       "1 1  ensembl_havana gene       11869 14412 .  +  . \n",
       "2 1  havana         transcript 11869 14409 .  +  . \n",
       "3 1  havana         exon       11869 12227 .  +  . \n",
       "4 1  havana         exon       12613 12721 .  +  . \n",
       "5 1  havana         exon       13221 14409 .  +  . \n",
       "6 1  ensembl        transcript 11872 14412 .  +  . \n",
       "  V9                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "1 gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";                                                                                                                                                                                                                                                                                                                 \n",
       "2 gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; tag \"basic\";                                                              \n",
       "3 gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"1\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002234944\"; exon_version \"1\"; tag \"basic\";\n",
       "4 gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00003582793\"; exon_version \"1\"; tag \"basic\";\n",
       "5 gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000456328\"; transcript_version \"2\"; exon_number \"3\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-002\"; transcript_source \"havana\"; transcript_biotype \"processed_transcript\"; havana_transcript \"OTTHUMT00000362751\"; havana_transcript_version \"1\"; exon_id \"ENSE00002312635\"; exon_version \"1\"; tag \"basic\";\n",
       "6 gene_id \"ENSG00000223972\"; gene_version \"4\"; transcript_id \"ENST00000515242\"; transcript_version \"2\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; transcript_name \"DDX11L1-201\"; transcript_source \"ensembl\"; transcript_biotype \"transcribed_unprocessed_pseudogene\";                                                                                                                                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene_coords)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Filter just for genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_coords_genesonly <- gene_coords %>%\n",
    "    filter(V3 == \"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>11869</td><td>14412</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";</td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>14363</td><td>29806</td><td>.</td><td>-</td><td>.</td><td>gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; </td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>29554</td><td>31109</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";</td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>34554</td><td>36081</td><td>.</td><td>-</td><td>.</td><td>gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";   </td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>52473</td><td>54936</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; </td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>gene</td><td>62948</td><td>63887</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & ensembl\\_havana & gene & 11869 & 14412 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\";\\\\\n",
       "\t 1 & ensembl\\_havana & gene & 14363 & 29806 & . & - & . & gene\\_id \"ENSG00000227232\"; gene\\_version \"4\"; gene\\_name \"WASH7P\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; \\\\\n",
       "\t 1 & ensembl\\_havana & gene & 29554 & 31109 & . & + & . & gene\\_id \"ENSG00000243485\"; gene\\_version \"2\"; gene\\_name \"MIR1302-10\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"lincRNA\";\\\\\n",
       "\t 1 & ensembl\\_havana & gene & 34554 & 36081 & . & - & . & gene\\_id \"ENSG00000237613\"; gene\\_version \"2\"; gene\\_name \"FAM138A\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"lincRNA\";   \\\\\n",
       "\t 1 & ensembl\\_havana & gene & 52473 & 54936 & . & + & . & gene\\_id \"ENSG00000268020\"; gene\\_version \"2\"; gene\\_name \"OR4G4P\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; \\\\\n",
       "\t 1 & havana         & gene & 62948 & 63887 & . & + & . & gene\\_id \"ENSG00000240361\"; gene\\_version \"1\"; gene\\_name \"OR4G11P\"; gene\\_source \"havana\"; gene\\_biotype \"pseudogene\";        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 9\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;chr&gt; | V3 &lt;chr&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ensembl_havana | gene | 11869 | 14412 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; |\n",
       "| 1 | ensembl_havana | gene | 14363 | 29806 | . | - | . | gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";  |\n",
       "| 1 | ensembl_havana | gene | 29554 | 31109 | . | + | . | gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\"; |\n",
       "| 1 | ensembl_havana | gene | 34554 | 36081 | . | - | . | gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";    |\n",
       "| 1 | ensembl_havana | gene | 52473 | 54936 | . | + | . | gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";  |\n",
       "| 1 | havana         | gene | 62948 | 63887 | . | + | . | gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";         |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2             V3   V4    V5    V6 V7 V8\n",
       "1 1  ensembl_havana gene 11869 14412 .  +  . \n",
       "2 1  ensembl_havana gene 14363 29806 .  -  . \n",
       "3 1  ensembl_havana gene 29554 31109 .  +  . \n",
       "4 1  ensembl_havana gene 34554 36081 .  -  . \n",
       "5 1  ensembl_havana gene 52473 54936 .  +  . \n",
       "6 1  havana         gene 62948 63887 .  +  . \n",
       "  V9                                                                                                                        \n",
       "1 gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";\n",
       "2 gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; \n",
       "3 gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";\n",
       "4 gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";   \n",
       "5 gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; \n",
       "6 gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene_coords_genesonly)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Separate column 9 to get gene names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "setDT(gene_coords_genesonly)[, c(\"gene_id\", \"gene_version\", \"gene_name\", \"gene_source\", \"gene_biotype\") := tstrsplit(V9, \";\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>gene_id</th><th scope=col>gene_version</th><th scope=col>gene_name</th><th scope=col>gene_source</th><th scope=col>gene_biotype</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>11869</td><td>14412</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";</td><td>gene_id \"ENSG00000223972\"</td><td> gene_version \"4\"</td><td> gene_name \"DDX11L1\"   </td><td> gene_source \"ensembl_havana\"</td><td> gene_biotype \"pseudogene\"</td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>14363</td><td>29806</td><td>.</td><td>-</td><td>.</td><td>gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; </td><td>gene_id \"ENSG00000227232\"</td><td> gene_version \"4\"</td><td> gene_name \"WASH7P\"    </td><td> gene_source \"ensembl_havana\"</td><td> gene_biotype \"pseudogene\"</td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>29554</td><td>31109</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";</td><td>gene_id \"ENSG00000243485\"</td><td> gene_version \"2\"</td><td> gene_name \"MIR1302-10\"</td><td> gene_source \"ensembl_havana\"</td><td> gene_biotype \"lincRNA\"   </td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>34554</td><td>36081</td><td>.</td><td>-</td><td>.</td><td>gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";   </td><td>gene_id \"ENSG00000237613\"</td><td> gene_version \"2\"</td><td> gene_name \"FAM138A\"   </td><td> gene_source \"ensembl_havana\"</td><td> gene_biotype \"lincRNA\"   </td></tr>\n",
       "\t<tr><td>1</td><td>ensembl_havana</td><td>gene</td><td>52473</td><td>54936</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; </td><td>gene_id \"ENSG00000268020\"</td><td> gene_version \"2\"</td><td> gene_name \"OR4G4P\"    </td><td> gene_source \"ensembl_havana\"</td><td> gene_biotype \"pseudogene\"</td></tr>\n",
       "\t<tr><td>1</td><td>havana        </td><td>gene</td><td>62948</td><td>63887</td><td>.</td><td>+</td><td>.</td><td>gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";        </td><td>gene_id \"ENSG00000240361\"</td><td> gene_version \"1\"</td><td> gene_name \"OR4G11P\"   </td><td> gene_source \"havana\"        </td><td> gene_biotype \"pseudogene\"</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 14\n",
       "\\begin{tabular}{llllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & gene\\_id & gene\\_version & gene\\_name & gene\\_source & gene\\_biotype\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & ensembl\\_havana & gene & 11869 & 14412 & . & + & . & gene\\_id \"ENSG00000223972\"; gene\\_version \"4\"; gene\\_name \"DDX11L1\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\"; & gene\\_id \"ENSG00000223972\" &  gene\\_version \"4\" &  gene\\_name \"DDX11L1\"    &  gene\\_source \"ensembl\\_havana\" &  gene\\_biotype \"pseudogene\"\\\\\n",
       "\t 1 & ensembl\\_havana & gene & 14363 & 29806 & . & - & . & gene\\_id \"ENSG00000227232\"; gene\\_version \"4\"; gene\\_name \"WASH7P\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\";  & gene\\_id \"ENSG00000227232\" &  gene\\_version \"4\" &  gene\\_name \"WASH7P\"     &  gene\\_source \"ensembl\\_havana\" &  gene\\_biotype \"pseudogene\"\\\\\n",
       "\t 1 & ensembl\\_havana & gene & 29554 & 31109 & . & + & . & gene\\_id \"ENSG00000243485\"; gene\\_version \"2\"; gene\\_name \"MIR1302-10\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"lincRNA\"; & gene\\_id \"ENSG00000243485\" &  gene\\_version \"2\" &  gene\\_name \"MIR1302-10\" &  gene\\_source \"ensembl\\_havana\" &  gene\\_biotype \"lincRNA\"   \\\\\n",
       "\t 1 & ensembl\\_havana & gene & 34554 & 36081 & . & - & . & gene\\_id \"ENSG00000237613\"; gene\\_version \"2\"; gene\\_name \"FAM138A\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"lincRNA\";    & gene\\_id \"ENSG00000237613\" &  gene\\_version \"2\" &  gene\\_name \"FAM138A\"    &  gene\\_source \"ensembl\\_havana\" &  gene\\_biotype \"lincRNA\"   \\\\\n",
       "\t 1 & ensembl\\_havana & gene & 52473 & 54936 & . & + & . & gene\\_id \"ENSG00000268020\"; gene\\_version \"2\"; gene\\_name \"OR4G4P\"; gene\\_source \"ensembl\\_havana\"; gene\\_biotype \"pseudogene\";  & gene\\_id \"ENSG00000268020\" &  gene\\_version \"2\" &  gene\\_name \"OR4G4P\"     &  gene\\_source \"ensembl\\_havana\" &  gene\\_biotype \"pseudogene\"\\\\\n",
       "\t 1 & havana         & gene & 62948 & 63887 & . & + & . & gene\\_id \"ENSG00000240361\"; gene\\_version \"1\"; gene\\_name \"OR4G11P\"; gene\\_source \"havana\"; gene\\_biotype \"pseudogene\";         & gene\\_id \"ENSG00000240361\" &  gene\\_version \"1\" &  gene\\_name \"OR4G11P\"    &  gene\\_source \"havana\"         &  gene\\_biotype \"pseudogene\"\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 14\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;chr&gt; | V3 &lt;chr&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;chr&gt; | gene_id &lt;chr&gt; | gene_version &lt;chr&gt; | gene_name &lt;chr&gt; | gene_source &lt;chr&gt; | gene_biotype &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ensembl_havana | gene | 11869 | 14412 | . | + | . | gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; | gene_id \"ENSG00000223972\" |  gene_version \"4\" |  gene_name \"DDX11L1\"    |  gene_source \"ensembl_havana\" |  gene_biotype \"pseudogene\" |\n",
       "| 1 | ensembl_havana | gene | 14363 | 29806 | . | - | . | gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";  | gene_id \"ENSG00000227232\" |  gene_version \"4\" |  gene_name \"WASH7P\"     |  gene_source \"ensembl_havana\" |  gene_biotype \"pseudogene\" |\n",
       "| 1 | ensembl_havana | gene | 29554 | 31109 | . | + | . | gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\"; | gene_id \"ENSG00000243485\" |  gene_version \"2\" |  gene_name \"MIR1302-10\" |  gene_source \"ensembl_havana\" |  gene_biotype \"lincRNA\"    |\n",
       "| 1 | ensembl_havana | gene | 34554 | 36081 | . | - | . | gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";    | gene_id \"ENSG00000237613\" |  gene_version \"2\" |  gene_name \"FAM138A\"    |  gene_source \"ensembl_havana\" |  gene_biotype \"lincRNA\"    |\n",
       "| 1 | ensembl_havana | gene | 52473 | 54936 | . | + | . | gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";  | gene_id \"ENSG00000268020\" |  gene_version \"2\" |  gene_name \"OR4G4P\"     |  gene_source \"ensembl_havana\" |  gene_biotype \"pseudogene\" |\n",
       "| 1 | havana         | gene | 62948 | 63887 | . | + | . | gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";         | gene_id \"ENSG00000240361\" |  gene_version \"1\" |  gene_name \"OR4G11P\"    |  gene_source \"havana\"         |  gene_biotype \"pseudogene\" |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2             V3   V4    V5    V6 V7 V8\n",
       "1 1  ensembl_havana gene 11869 14412 .  +  . \n",
       "2 1  ensembl_havana gene 14363 29806 .  -  . \n",
       "3 1  ensembl_havana gene 29554 31109 .  +  . \n",
       "4 1  ensembl_havana gene 34554 36081 .  -  . \n",
       "5 1  ensembl_havana gene 52473 54936 .  +  . \n",
       "6 1  havana         gene 62948 63887 .  +  . \n",
       "  V9                                                                                                                        \n",
       "1 gene_id \"ENSG00000223972\"; gene_version \"4\"; gene_name \"DDX11L1\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\";\n",
       "2 gene_id \"ENSG00000227232\"; gene_version \"4\"; gene_name \"WASH7P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; \n",
       "3 gene_id \"ENSG00000243485\"; gene_version \"2\"; gene_name \"MIR1302-10\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";\n",
       "4 gene_id \"ENSG00000237613\"; gene_version \"2\"; gene_name \"FAM138A\"; gene_source \"ensembl_havana\"; gene_biotype \"lincRNA\";   \n",
       "5 gene_id \"ENSG00000268020\"; gene_version \"2\"; gene_name \"OR4G4P\"; gene_source \"ensembl_havana\"; gene_biotype \"pseudogene\"; \n",
       "6 gene_id \"ENSG00000240361\"; gene_version \"1\"; gene_name \"OR4G11P\"; gene_source \"havana\"; gene_biotype \"pseudogene\";        \n",
       "  gene_id                   gene_version      gene_name              \n",
       "1 gene_id \"ENSG00000223972\"  gene_version \"4\"  gene_name \"DDX11L1\"   \n",
       "2 gene_id \"ENSG00000227232\"  gene_version \"4\"  gene_name \"WASH7P\"    \n",
       "3 gene_id \"ENSG00000243485\"  gene_version \"2\"  gene_name \"MIR1302-10\"\n",
       "4 gene_id \"ENSG00000237613\"  gene_version \"2\"  gene_name \"FAM138A\"   \n",
       "5 gene_id \"ENSG00000268020\"  gene_version \"2\"  gene_name \"OR4G4P\"    \n",
       "6 gene_id \"ENSG00000240361\"  gene_version \"1\"  gene_name \"OR4G11P\"   \n",
       "  gene_source                   gene_biotype              \n",
       "1  gene_source \"ensembl_havana\"  gene_biotype \"pseudogene\"\n",
       "2  gene_source \"ensembl_havana\"  gene_biotype \"pseudogene\"\n",
       "3  gene_source \"ensembl_havana\"  gene_biotype \"lincRNA\"   \n",
       "4  gene_source \"ensembl_havana\"  gene_biotype \"lincRNA\"   \n",
       "5  gene_source \"ensembl_havana\"  gene_biotype \"pseudogene\"\n",
       "6  gene_source \"havana\"          gene_biotype \"pseudogene\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene_coords_genesonly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_coords_genesonly_select <- gene_coords_genesonly %>%\n",
    "    select(V1, V4, V5, gene_name) %>% #Select just relevant columns - chr, bp start, bp end, gene name\n",
    "    mutate(gene = gsub(\"gene_name\", \"\", gene_name)) %>% #Remove string 'gene_name' from the gene_name col\n",
    "    mutate(gene_final = gsub('\"', \"\", gene)) #Remove the \"\" from gene column and make gene_final col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>gene_name</th><th scope=col>gene</th><th scope=col>gene_final</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>11869</td><td>14412</td><td> gene_name \"DDX11L1\"   </td><td>  \"DDX11L1\"   </td><td>  DDX11L1   </td></tr>\n",
       "\t<tr><td>1</td><td>14363</td><td>29806</td><td> gene_name \"WASH7P\"    </td><td>  \"WASH7P\"    </td><td>  WASH7P    </td></tr>\n",
       "\t<tr><td>1</td><td>29554</td><td>31109</td><td> gene_name \"MIR1302-10\"</td><td>  \"MIR1302-10\"</td><td>  MIR1302-10</td></tr>\n",
       "\t<tr><td>1</td><td>34554</td><td>36081</td><td> gene_name \"FAM138A\"   </td><td>  \"FAM138A\"   </td><td>  FAM138A   </td></tr>\n",
       "\t<tr><td>1</td><td>52473</td><td>54936</td><td> gene_name \"OR4G4P\"    </td><td>  \"OR4G4P\"    </td><td>  OR4G4P    </td></tr>\n",
       "\t<tr><td>1</td><td>62948</td><td>63887</td><td> gene_name \"OR4G11P\"   </td><td>  \"OR4G11P\"   </td><td>  OR4G11P   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " V1 & V4 & V5 & gene\\_name & gene & gene\\_final\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & 11869 & 14412 &  gene\\_name \"DDX11L1\"    &   \"DDX11L1\"    &   DDX11L1   \\\\\n",
       "\t 1 & 14363 & 29806 &  gene\\_name \"WASH7P\"     &   \"WASH7P\"     &   WASH7P    \\\\\n",
       "\t 1 & 29554 & 31109 &  gene\\_name \"MIR1302-10\" &   \"MIR1302-10\" &   MIR1302-10\\\\\n",
       "\t 1 & 34554 & 36081 &  gene\\_name \"FAM138A\"    &   \"FAM138A\"    &   FAM138A   \\\\\n",
       "\t 1 & 52473 & 54936 &  gene\\_name \"OR4G4P\"     &   \"OR4G4P\"     &   OR4G4P    \\\\\n",
       "\t 1 & 62948 & 63887 &  gene\\_name \"OR4G11P\"    &   \"OR4G11P\"    &   OR4G11P   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 6\n",
       "\n",
       "| V1 &lt;chr&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | gene_name &lt;chr&gt; | gene &lt;chr&gt; | gene_final &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | 11869 | 14412 |  gene_name \"DDX11L1\"    |   \"DDX11L1\"    |   DDX11L1    |\n",
       "| 1 | 14363 | 29806 |  gene_name \"WASH7P\"     |   \"WASH7P\"     |   WASH7P     |\n",
       "| 1 | 29554 | 31109 |  gene_name \"MIR1302-10\" |   \"MIR1302-10\" |   MIR1302-10 |\n",
       "| 1 | 34554 | 36081 |  gene_name \"FAM138A\"    |   \"FAM138A\"    |   FAM138A    |\n",
       "| 1 | 52473 | 54936 |  gene_name \"OR4G4P\"     |   \"OR4G4P\"     |   OR4G4P     |\n",
       "| 1 | 62948 | 63887 |  gene_name \"OR4G11P\"    |   \"OR4G11P\"    |   OR4G11P    |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V4    V5    gene_name               gene           gene_final  \n",
       "1 1  11869 14412  gene_name \"DDX11L1\"      \"DDX11L1\"      DDX11L1   \n",
       "2 1  14363 29806  gene_name \"WASH7P\"       \"WASH7P\"       WASH7P    \n",
       "3 1  29554 31109  gene_name \"MIR1302-10\"   \"MIR1302-10\"   MIR1302-10\n",
       "4 1  34554 36081  gene_name \"FAM138A\"      \"FAM138A\"      FAM138A   \n",
       "5 1  52473 54936  gene_name \"OR4G4P\"       \"OR4G4P\"       OR4G4P    \n",
       "6 1  62948 63887  gene_name \"OR4G11P\"      \"OR4G11P\"      OR4G11P   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene_coords_genesonly_select)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select only relevant columns and rename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_coords_final <- gene_coords_genesonly_select %>%\n",
    "    select(V1, V4, V5, gene_final) %>%\n",
    "    rename(chr = V1,\n",
    "          bp_start = V4,\n",
    "          bp_end = V5,\n",
    "          gene = gene_final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>bp_start</th><th scope=col>bp_end</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>11869</td><td>14412</td><td>  DDX11L1   </td></tr>\n",
       "\t<tr><td>1</td><td>14363</td><td>29806</td><td>  WASH7P    </td></tr>\n",
       "\t<tr><td>1</td><td>29554</td><td>31109</td><td>  MIR1302-10</td></tr>\n",
       "\t<tr><td>1</td><td>34554</td><td>36081</td><td>  FAM138A   </td></tr>\n",
       "\t<tr><td>1</td><td>52473</td><td>54936</td><td>  OR4G4P    </td></tr>\n",
       "\t<tr><td>1</td><td>62948</td><td>63887</td><td>  OR4G11P   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " chr & bp\\_start & bp\\_end & gene\\\\\n",
       " <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & 11869 & 14412 &   DDX11L1   \\\\\n",
       "\t 1 & 14363 & 29806 &   WASH7P    \\\\\n",
       "\t 1 & 29554 & 31109 &   MIR1302-10\\\\\n",
       "\t 1 & 34554 & 36081 &   FAM138A   \\\\\n",
       "\t 1 & 52473 & 54936 &   OR4G4P    \\\\\n",
       "\t 1 & 62948 & 63887 &   OR4G11P   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 4\n",
       "\n",
       "| chr &lt;chr&gt; | bp_start &lt;int&gt; | bp_end &lt;int&gt; | gene &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | 11869 | 14412 |   DDX11L1    |\n",
       "| 1 | 14363 | 29806 |   WASH7P     |\n",
       "| 1 | 29554 | 31109 |   MIR1302-10 |\n",
       "| 1 | 34554 | 36081 |   FAM138A    |\n",
       "| 1 | 52473 | 54936 |   OR4G4P     |\n",
       "| 1 | 62948 | 63887 |   OR4G11P    |\n",
       "\n"
      ],
      "text/plain": [
       "  chr bp_start bp_end gene        \n",
       "1 1   11869    14412    DDX11L1   \n",
       "2 1   14363    29806    WASH7P    \n",
       "3 1   29554    31109    MIR1302-10\n",
       "4 1   34554    36081    FAM138A   \n",
       "5 1   52473    54936    OR4G4P    \n",
       "6 1   62948    63887    OR4G11P   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(gene_coords_final)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate GWAS sumstats with nearest genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>freq</th><th scope=col>b</th><th scope=col>se</th><th scope=col>p</th><th scope=col>N_cases</th><th scope=col>N_controls</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr5:60345424 </td><td>T</td><td>G</td><td>0.8883</td><td>-0.1916</td><td>0.0266</td><td>5.618e-13</td><td>33674</td><td>449056</td></tr>\n",
       "\t<tr><td>chr17:43758382</td><td>T</td><td>C</td><td>0.7814</td><td> 0.2527</td><td>0.0269</td><td>5.906e-21</td><td>32505</td><td>448088</td></tr>\n",
       "\t<tr><td>chr17:44201667</td><td>A</td><td>G</td><td>0.2176</td><td>-0.2554</td><td>0.0271</td><td>4.138e-21</td><td>32505</td><td>448088</td></tr>\n",
       "\t<tr><td>chr12:40556767</td><td>T</td><td>C</td><td>0.9971</td><td>-4.0052</td><td>0.5940</td><td>1.554e-11</td><td> 8488</td><td>  7671</td></tr>\n",
       "\t<tr><td>chr17:43858187</td><td>T</td><td>G</td><td>0.2187</td><td>-0.2521</td><td>0.0270</td><td>8.896e-21</td><td>32505</td><td>448088</td></tr>\n",
       "\t<tr><td>chr17:43992233</td><td>T</td><td>C</td><td>0.2171</td><td>-0.2510</td><td>0.0271</td><td>1.749e-20</td><td>32505</td><td>448088</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " SNP & A1 & A2 & freq & b & se & p & N\\_cases & N\\_controls\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr5:60345424  & T & G & 0.8883 & -0.1916 & 0.0266 & 5.618e-13 & 33674 & 449056\\\\\n",
       "\t chr17:43758382 & T & C & 0.7814 &  0.2527 & 0.0269 & 5.906e-21 & 32505 & 448088\\\\\n",
       "\t chr17:44201667 & A & G & 0.2176 & -0.2554 & 0.0271 & 4.138e-21 & 32505 & 448088\\\\\n",
       "\t chr12:40556767 & T & C & 0.9971 & -4.0052 & 0.5940 & 1.554e-11 &  8488 &   7671\\\\\n",
       "\t chr17:43858187 & T & G & 0.2187 & -0.2521 & 0.0270 & 8.896e-21 & 32505 & 448088\\\\\n",
       "\t chr17:43992233 & T & C & 0.2171 & -0.2510 & 0.0271 & 1.749e-20 & 32505 & 448088\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 9\n",
       "\n",
       "| SNP &lt;chr&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | freq &lt;dbl&gt; | b &lt;dbl&gt; | se &lt;dbl&gt; | p &lt;dbl&gt; | N_cases &lt;int&gt; | N_controls &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| chr5:60345424  | T | G | 0.8883 | -0.1916 | 0.0266 | 5.618e-13 | 33674 | 449056 |\n",
       "| chr17:43758382 | T | C | 0.7814 |  0.2527 | 0.0269 | 5.906e-21 | 32505 | 448088 |\n",
       "| chr17:44201667 | A | G | 0.2176 | -0.2554 | 0.0271 | 4.138e-21 | 32505 | 448088 |\n",
       "| chr12:40556767 | T | C | 0.9971 | -4.0052 | 0.5940 | 1.554e-11 |  8488 |   7671 |\n",
       "| chr17:43858187 | T | G | 0.2187 | -0.2521 | 0.0270 | 8.896e-21 | 32505 | 448088 |\n",
       "| chr17:43992233 | T | C | 0.2171 | -0.2510 | 0.0271 | 1.749e-20 | 32505 | 448088 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP            A1 A2 freq   b       se     p         N_cases N_controls\n",
       "1 chr5:60345424  T  G  0.8883 -0.1916 0.0266 5.618e-13 33674   449056    \n",
       "2 chr17:43758382 T  C  0.7814  0.2527 0.0269 5.906e-21 32505   448088    \n",
       "3 chr17:44201667 A  G  0.2176 -0.2554 0.0271 4.138e-21 32505   448088    \n",
       "4 chr12:40556767 T  C  0.9971 -4.0052 0.5940 1.554e-11  8488     7671    \n",
       "5 chr17:43858187 T  G  0.2187 -0.2521 0.0270 8.896e-21 32505   448088    \n",
       "6 chr17:43992233 T  C  0.2171 -0.2510 0.0271 1.749e-20 32505   448088    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail(data_filtered)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in GWAS top hits annotated with nearest genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First import file into workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Executing:  gsutil -mu gp2-ipdgc-hackathon cp gs://fc-secure-b9f9d17f-b38c-407d-85e0-7a759f13cea0/\\\"Table S2. Detailed summary statistics.xlsx\\\" ~/bin/data_temp/\"\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shell_do(str_glue('gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/\"Table S2. Detailed summary statistics.xlsx\" ~/bin/data_temp/'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'nallsEtAl2019_excluding23andMe_allVariants.tab'</li><li>'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'</li><li>'NCBI37.3.zip'</li><li>'PDGWAS_for_FUMA_MT.txt'</li><li>'PDGWAS_for_FUMA_MT.txt.zip'</li><li>'Table S2. Detailed summary statistics.xlsx'</li><li>'test.zip'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab'\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab.zip'\n",
       "\\item 'NCBI37.3.zip'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt.zip'\n",
       "\\item 'Table S2. Detailed summary statistics.xlsx'\n",
       "\\item 'test.zip'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'nallsEtAl2019_excluding23andMe_allVariants.tab'\n",
       "2. 'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'\n",
       "3. 'NCBI37.3.zip'\n",
       "4. 'PDGWAS_for_FUMA_MT.txt'\n",
       "5. 'PDGWAS_for_FUMA_MT.txt.zip'\n",
       "6. 'Table S2. Detailed summary statistics.xlsx'\n",
       "7. 'test.zip'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"nallsEtAl2019_excluding23andMe_allVariants.tab\"    \n",
       "[2] \"nallsEtAl2019_excluding23andMe_allVariants.tab.zip\"\n",
       "[3] \"NCBI37.3.zip\"                                      \n",
       "[4] \"PDGWAS_for_FUMA_MT.txt\"                            \n",
       "[5] \"PDGWAS_for_FUMA_MT.txt.zip\"                        \n",
       "[6] \"Table S2. Detailed summary statistics.xlsx\"        \n",
       "[7] \"test.zip\"                                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('ls ~/bin/data_temp', intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read GWAS hits into R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "GWAS_toploci <- read_xlsx(\"~/bin/data_temp/Table S2. Detailed summary statistics.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 31</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>CHR</th><th scope=col>BP</th><th scope=col>Nearest Gene</th><th scope=col>QTL Nominated Gene (nearest QTL)</th><th scope=col>Effect allele</th><th scope=col>Other allele</th><th scope=col>Effect allele frequency</th><th scope=col>Beta, all studies</th><th scope=col>SE, all studies</th><th scope=col>⋯</th><th scope=col>Freq1, new studies</th><th scope=col>Beta, new studies</th><th scope=col>StdErr, new studies</th><th scope=col>P, new studies</th><th scope=col>I2, new studies</th><th scope=col>Passes pooled 23andMe QC</th><th scope=col>Known GWAS locus within 1MB</th><th scope=col>Failed final filtering and QC</th><th scope=col>Locus within 250KB</th><th scope=col>Locus Number</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>rs114138760</td><td>1</td><td>154898185</td><td>PMVK   </td><td>NA    </td><td>c</td><td>g</td><td>0.0112</td><td> 0.2812</td><td>0.0478</td><td>⋯</td><td>0.0109</td><td> 0.1997</td><td>0.0843</td><td>1.779e-02</td><td> 0.0</td><td>T</td><td>1</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>rs35749011 </td><td>1</td><td>155135036</td><td>KRTCAP2</td><td>NA    </td><td>a</td><td>g</td><td>0.0169</td><td> 0.6068</td><td>0.0342</td><td>⋯</td><td>0.0177</td><td> 0.6798</td><td>0.0615</td><td>2.160e-28</td><td> 0.0</td><td>T</td><td>1</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>rs76763715 </td><td>1</td><td>155205634</td><td>GBAP1  </td><td>GBAP1 </td><td>t</td><td>c</td><td>0.9953</td><td>-0.7467</td><td>0.0765</td><td>⋯</td><td>0.9941</td><td>-0.6693</td><td>0.1269</td><td>1.340e-07</td><td> 0.0</td><td>F</td><td>1</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>rs6658353  </td><td>1</td><td>161469054</td><td>FCGR2A </td><td>FCGR2A</td><td>c</td><td>g</td><td>0.5011</td><td> 0.0650</td><td>0.0094</td><td>⋯</td><td>0.5105</td><td> 0.0662</td><td>0.0171</td><td>1.101e-04</td><td>46.4</td><td>T</td><td>0</td><td>0</td><td>0</td><td>2</td></tr>\n",
       "\t<tr><td>rs11578699 </td><td>1</td><td>171719769</td><td>VAMP4  </td><td>VAMP4 </td><td>t</td><td>c</td><td>0.1949</td><td>-0.0704</td><td>0.0120</td><td>⋯</td><td>0.1947</td><td>-0.0663</td><td>0.0226</td><td>3.313e-03</td><td>13.4</td><td>T</td><td>0</td><td>0</td><td>0</td><td>3</td></tr>\n",
       "\t<tr><td>rs823118   </td><td>1</td><td>205723572</td><td>NUCKS1 </td><td>NUCKS1</td><td>t</td><td>c</td><td>0.5660</td><td> 0.1066</td><td>0.0094</td><td>⋯</td><td>0.5674</td><td> 0.0842</td><td>0.0172</td><td>1.020e-06</td><td> 0.0</td><td>T</td><td>1</td><td>0</td><td>1</td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 31\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " SNP & CHR & BP & Nearest Gene & QTL Nominated Gene (nearest QTL) & Effect allele & Other allele & Effect allele frequency & Beta, all studies & SE, all studies & ⋯ & Freq1, new studies & Beta, new studies & StdErr, new studies & P, new studies & I2, new studies & Passes pooled 23andMe QC & Known GWAS locus within 1MB & Failed final filtering and QC & Locus within 250KB & Locus Number\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t rs114138760 & 1 & 154898185 & PMVK    & NA     & c & g & 0.0112 &  0.2812 & 0.0478 & ⋯ & 0.0109 &  0.1997 & 0.0843 & 1.779e-02 &  0.0 & T & 1 & 0 & 1 & 1\\\\\n",
       "\t rs35749011  & 1 & 155135036 & KRTCAP2 & NA     & a & g & 0.0169 &  0.6068 & 0.0342 & ⋯ & 0.0177 &  0.6798 & 0.0615 & 2.160e-28 &  0.0 & T & 1 & 0 & 1 & 1\\\\\n",
       "\t rs76763715  & 1 & 155205634 & GBAP1   & GBAP1  & t & c & 0.9953 & -0.7467 & 0.0765 & ⋯ & 0.9941 & -0.6693 & 0.1269 & 1.340e-07 &  0.0 & F & 1 & 0 & 1 & 1\\\\\n",
       "\t rs6658353   & 1 & 161469054 & FCGR2A  & FCGR2A & c & g & 0.5011 &  0.0650 & 0.0094 & ⋯ & 0.5105 &  0.0662 & 0.0171 & 1.101e-04 & 46.4 & T & 0 & 0 & 0 & 2\\\\\n",
       "\t rs11578699  & 1 & 171719769 & VAMP4   & VAMP4  & t & c & 0.1949 & -0.0704 & 0.0120 & ⋯ & 0.1947 & -0.0663 & 0.0226 & 3.313e-03 & 13.4 & T & 0 & 0 & 0 & 3\\\\\n",
       "\t rs823118    & 1 & 205723572 & NUCKS1  & NUCKS1 & t & c & 0.5660 &  0.1066 & 0.0094 & ⋯ & 0.5674 &  0.0842 & 0.0172 & 1.020e-06 &  0.0 & T & 1 & 0 & 1 & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 31\n",
       "\n",
       "| SNP &lt;chr&gt; | CHR &lt;dbl&gt; | BP &lt;dbl&gt; | Nearest Gene &lt;chr&gt; | QTL Nominated Gene (nearest QTL) &lt;chr&gt; | Effect allele &lt;chr&gt; | Other allele &lt;chr&gt; | Effect allele frequency &lt;dbl&gt; | Beta, all studies &lt;dbl&gt; | SE, all studies &lt;dbl&gt; | ⋯ ⋯ | Freq1, new studies &lt;dbl&gt; | Beta, new studies &lt;dbl&gt; | StdErr, new studies &lt;dbl&gt; | P, new studies &lt;dbl&gt; | I2, new studies &lt;dbl&gt; | Passes pooled 23andMe QC &lt;chr&gt; | Known GWAS locus within 1MB &lt;dbl&gt; | Failed final filtering and QC &lt;dbl&gt; | Locus within 250KB &lt;chr&gt; | Locus Number &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| rs114138760 | 1 | 154898185 | PMVK    | NA     | c | g | 0.0112 |  0.2812 | 0.0478 | ⋯ | 0.0109 |  0.1997 | 0.0843 | 1.779e-02 |  0.0 | T | 1 | 0 | 1 | 1 |\n",
       "| rs35749011  | 1 | 155135036 | KRTCAP2 | NA     | a | g | 0.0169 |  0.6068 | 0.0342 | ⋯ | 0.0177 |  0.6798 | 0.0615 | 2.160e-28 |  0.0 | T | 1 | 0 | 1 | 1 |\n",
       "| rs76763715  | 1 | 155205634 | GBAP1   | GBAP1  | t | c | 0.9953 | -0.7467 | 0.0765 | ⋯ | 0.9941 | -0.6693 | 0.1269 | 1.340e-07 |  0.0 | F | 1 | 0 | 1 | 1 |\n",
       "| rs6658353   | 1 | 161469054 | FCGR2A  | FCGR2A | c | g | 0.5011 |  0.0650 | 0.0094 | ⋯ | 0.5105 |  0.0662 | 0.0171 | 1.101e-04 | 46.4 | T | 0 | 0 | 0 | 2 |\n",
       "| rs11578699  | 1 | 171719769 | VAMP4   | VAMP4  | t | c | 0.1949 | -0.0704 | 0.0120 | ⋯ | 0.1947 | -0.0663 | 0.0226 | 3.313e-03 | 13.4 | T | 0 | 0 | 0 | 3 |\n",
       "| rs823118    | 1 | 205723572 | NUCKS1  | NUCKS1 | t | c | 0.5660 |  0.1066 | 0.0094 | ⋯ | 0.5674 |  0.0842 | 0.0172 | 1.020e-06 |  0.0 | T | 1 | 0 | 1 | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP         CHR BP        Nearest Gene QTL Nominated Gene (nearest QTL)\n",
       "1 rs114138760 1   154898185 PMVK         NA                              \n",
       "2 rs35749011  1   155135036 KRTCAP2      NA                              \n",
       "3 rs76763715  1   155205634 GBAP1        GBAP1                           \n",
       "4 rs6658353   1   161469054 FCGR2A       FCGR2A                          \n",
       "5 rs11578699  1   171719769 VAMP4        VAMP4                           \n",
       "6 rs823118    1   205723572 NUCKS1       NUCKS1                          \n",
       "  Effect allele Other allele Effect allele frequency Beta, all studies\n",
       "1 c             g            0.0112                   0.2812          \n",
       "2 a             g            0.0169                   0.6068          \n",
       "3 t             c            0.9953                  -0.7467          \n",
       "4 c             g            0.5011                   0.0650          \n",
       "5 t             c            0.1949                  -0.0704          \n",
       "6 t             c            0.5660                   0.1066          \n",
       "  SE, all studies ⋯ Freq1, new studies Beta, new studies StdErr, new studies\n",
       "1 0.0478          ⋯ 0.0109              0.1997           0.0843             \n",
       "2 0.0342          ⋯ 0.0177              0.6798           0.0615             \n",
       "3 0.0765          ⋯ 0.9941             -0.6693           0.1269             \n",
       "4 0.0094          ⋯ 0.5105              0.0662           0.0171             \n",
       "5 0.0120          ⋯ 0.1947             -0.0663           0.0226             \n",
       "6 0.0094          ⋯ 0.5674              0.0842           0.0172             \n",
       "  P, new studies I2, new studies Passes pooled 23andMe QC\n",
       "1 1.779e-02       0.0            T                       \n",
       "2 2.160e-28       0.0            T                       \n",
       "3 1.340e-07       0.0            F                       \n",
       "4 1.101e-04      46.4            T                       \n",
       "5 3.313e-03      13.4            T                       \n",
       "6 1.020e-06       0.0            T                       \n",
       "  Known GWAS locus within 1MB Failed final filtering and QC Locus within 250KB\n",
       "1 1                           0                             1                 \n",
       "2 1                           0                             1                 \n",
       "3 1                           0                             1                 \n",
       "4 0                           0                             0                 \n",
       "5 0                           0                             0                 \n",
       "6 1                           0                             1                 \n",
       "  Locus Number\n",
       "1 1           \n",
       "2 1           \n",
       "3 1           \n",
       "4 2           \n",
       "5 3           \n",
       "6 4           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(GWAS_toploci)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make list of SNPs in .txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "SNPs_only <- GWAS_toploci %>%\n",
    "    select(SNP) %>%\n",
    "    arrange('P, all studies')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(SNPs_only, \"~/bin/data_temp/PDSNPs_sorted.txt\", quote = F, row.names = F, col.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'nallsEtAl2019_excluding23andMe_allVariants.tab'</li><li>'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'</li><li>'NCBI37.3.zip'</li><li>'PDgenes_sorted_geneonly.txt'</li><li>'PDGWAS_for_FUMA_MT.txt'</li><li>'PDGWAS_for_FUMA_MT.txt.zip'</li><li>'PDSNPs_sorted.txt'</li><li>'Table S2. Detailed summary statistics.xlsx'</li><li>'test.zip'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab'\n",
       "\\item 'nallsEtAl2019\\_excluding23andMe\\_allVariants.tab.zip'\n",
       "\\item 'NCBI37.3.zip'\n",
       "\\item 'PDgenes\\_sorted\\_geneonly.txt'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt'\n",
       "\\item 'PDGWAS\\_for\\_FUMA\\_MT.txt.zip'\n",
       "\\item 'PDSNPs\\_sorted.txt'\n",
       "\\item 'Table S2. Detailed summary statistics.xlsx'\n",
       "\\item 'test.zip'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'nallsEtAl2019_excluding23andMe_allVariants.tab'\n",
       "2. 'nallsEtAl2019_excluding23andMe_allVariants.tab.zip'\n",
       "3. 'NCBI37.3.zip'\n",
       "4. 'PDgenes_sorted_geneonly.txt'\n",
       "5. 'PDGWAS_for_FUMA_MT.txt'\n",
       "6. 'PDGWAS_for_FUMA_MT.txt.zip'\n",
       "7. 'PDSNPs_sorted.txt'\n",
       "8. 'Table S2. Detailed summary statistics.xlsx'\n",
       "9. 'test.zip'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"nallsEtAl2019_excluding23andMe_allVariants.tab\"    \n",
       "[2] \"nallsEtAl2019_excluding23andMe_allVariants.tab.zip\"\n",
       "[3] \"NCBI37.3.zip\"                                      \n",
       "[4] \"PDgenes_sorted_geneonly.txt\"                       \n",
       "[5] \"PDGWAS_for_FUMA_MT.txt\"                            \n",
       "[6] \"PDGWAS_for_FUMA_MT.txt.zip\"                        \n",
       "[7] \"PDSNPs_sorted.txt\"                                 \n",
       "[8] \"Table S2. Detailed summary statistics.xlsx\"        \n",
       "[9] \"test.zip\"                                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('ls ~/bin/data_temp/', intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make gene list ranked by p-values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "PDgenes <- GWAS_toploci %>%\n",
    "    select('Nearest Gene', 'P, all studies') %>%\n",
    "    rename(gene = 'Nearest Gene',\n",
    "          p = 'P, all studies')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>p</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>PMVK   </td><td>4.19e-09</td></tr>\n",
       "\t<tr><td>KRTCAP2</td><td>1.72e-70</td></tr>\n",
       "\t<tr><td>GBAP1  </td><td>1.59e-22</td></tr>\n",
       "\t<tr><td>FCGR2A </td><td>6.10e-12</td></tr>\n",
       "\t<tr><td>VAMP4  </td><td>4.47e-09</td></tr>\n",
       "\t<tr><td>NUCKS1 </td><td>1.11e-29</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " gene & p\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t PMVK    & 4.19e-09\\\\\n",
       "\t KRTCAP2 & 1.72e-70\\\\\n",
       "\t GBAP1   & 1.59e-22\\\\\n",
       "\t FCGR2A  & 6.10e-12\\\\\n",
       "\t VAMP4   & 4.47e-09\\\\\n",
       "\t NUCKS1  & 1.11e-29\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| gene &lt;chr&gt; | p &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| PMVK    | 4.19e-09 |\n",
       "| KRTCAP2 | 1.72e-70 |\n",
       "| GBAP1   | 1.59e-22 |\n",
       "| FCGR2A  | 6.10e-12 |\n",
       "| VAMP4   | 4.47e-09 |\n",
       "| NUCKS1  | 1.11e-29 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene    p       \n",
       "1 PMVK    4.19e-09\n",
       "2 KRTCAP2 1.72e-70\n",
       "3 GBAP1   1.59e-22\n",
       "4 FCGR2A  6.10e-12\n",
       "5 VAMP4   4.47e-09\n",
       "6 NUCKS1  1.11e-29"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(PDgenes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove any missing nearest gene and rank by p-value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "PDgenes_sorted <- PDgenes %>%\n",
    "    filter(!is.na(gene)) %>%\n",
    "    arrange(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>p</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SNCA   </td><td>3.89e-154</td></tr>\n",
       "\t<tr><td>SNCA   </td><td>5.16e-149</td></tr>\n",
       "\t<tr><td>LRRK2  </td><td>3.61e-148</td></tr>\n",
       "\t<tr><td>KRTCAP2</td><td> 1.72e-70</td></tr>\n",
       "\t<tr><td>TMEM175</td><td> 9.98e-70</td></tr>\n",
       "\t<tr><td>CRHR1  </td><td> 3.58e-68</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " gene & p\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t SNCA    & 3.89e-154\\\\\n",
       "\t SNCA    & 5.16e-149\\\\\n",
       "\t LRRK2   & 3.61e-148\\\\\n",
       "\t KRTCAP2 &  1.72e-70\\\\\n",
       "\t TMEM175 &  9.98e-70\\\\\n",
       "\t CRHR1   &  3.58e-68\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| gene &lt;chr&gt; | p &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| SNCA    | 3.89e-154 |\n",
       "| SNCA    | 5.16e-149 |\n",
       "| LRRK2   | 3.61e-148 |\n",
       "| KRTCAP2 |  1.72e-70 |\n",
       "| TMEM175 |  9.98e-70 |\n",
       "| CRHR1   |  3.58e-68 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene    p        \n",
       "1 SNCA    3.89e-154\n",
       "2 SNCA    5.16e-149\n",
       "3 LRRK2   3.61e-148\n",
       "4 KRTCAP2  1.72e-70\n",
       "5 TMEM175  9.98e-70\n",
       "6 CRHR1    3.58e-68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(PDgenes_sorted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(PDgenes_sorted, \"~/bin/data_temp/PDgenes_sorted_pvals.rnk\",\n",
    "           quote = F, col.names = F, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a dataframe with genes and -log10 pvalues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "PDgenes_sorted_neglogp <- PDgenes_sorted %>%\n",
    "    mutate(neg_log10p = -log10(p))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>p</th><th scope=col>neg_log10p</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SNCA   </td><td>3.89e-154</td><td>153.41005</td></tr>\n",
       "\t<tr><td>SNCA   </td><td>5.16e-149</td><td>148.28735</td></tr>\n",
       "\t<tr><td>LRRK2  </td><td>3.61e-148</td><td>147.44249</td></tr>\n",
       "\t<tr><td>KRTCAP2</td><td> 1.72e-70</td><td> 69.76447</td></tr>\n",
       "\t<tr><td>TMEM175</td><td> 9.98e-70</td><td> 69.00087</td></tr>\n",
       "\t<tr><td>CRHR1  </td><td> 3.58e-68</td><td> 67.44612</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " gene & p & neg\\_log10p\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t SNCA    & 3.89e-154 & 153.41005\\\\\n",
       "\t SNCA    & 5.16e-149 & 148.28735\\\\\n",
       "\t LRRK2   & 3.61e-148 & 147.44249\\\\\n",
       "\t KRTCAP2 &  1.72e-70 &  69.76447\\\\\n",
       "\t TMEM175 &  9.98e-70 &  69.00087\\\\\n",
       "\t CRHR1   &  3.58e-68 &  67.44612\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 3\n",
       "\n",
       "| gene &lt;chr&gt; | p &lt;dbl&gt; | neg_log10p &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| SNCA    | 3.89e-154 | 153.41005 |\n",
       "| SNCA    | 5.16e-149 | 148.28735 |\n",
       "| LRRK2   | 3.61e-148 | 147.44249 |\n",
       "| KRTCAP2 |  1.72e-70 |  69.76447 |\n",
       "| TMEM175 |  9.98e-70 |  69.00087 |\n",
       "| CRHR1   |  3.58e-68 |  67.44612 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene    p         neg_log10p\n",
       "1 SNCA    3.89e-154 153.41005 \n",
       "2 SNCA    5.16e-149 148.28735 \n",
       "3 LRRK2   3.61e-148 147.44249 \n",
       "4 KRTCAP2  1.72e-70  69.76447 \n",
       "5 TMEM175  9.98e-70  69.00087 \n",
       "6 CRHR1    3.58e-68  67.44612 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(PDgenes_sorted_neglogp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "PDgenes_sorted_neglogp <- PDgenes_sorted_neglogp %>%\n",
    "    select(gene, neg_log10p)\n",
    "\n",
    "#Export as rnk file\n",
    "write.table(PDgenes_sorted_neglogp, \"~/bin/data_temp/PDgenes_sorted_pvals_neglog.rnk\",\n",
    "           quote = F, col.names = F, row.names = F, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a dataframe with just the unique gene names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "PDgenes_sorted_geneonly <- PDgenes_sorted %>%\n",
    "    select(gene) %>%\n",
    "    distinct(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(PDgenes_sorted_geneonly, \"~/bin/data_temp/PDgenes_sorted_geneonly.txt\",\n",
    "           quote = F, col.names = F, row.names = F, sep = \"/n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SNCA'</li><li>'LRRK2'</li><li>'KRTCAP2'</li><li>'TMEM175'</li><li>'CRHR1'</li><li>'NSF'</li><li>'KANSL1'</li><li>'STK39'</li><li>'MAPT-AS1'</li><li>'HIP1R'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SNCA'\n",
       "\\item 'LRRK2'\n",
       "\\item 'KRTCAP2'\n",
       "\\item 'TMEM175'\n",
       "\\item 'CRHR1'\n",
       "\\item 'NSF'\n",
       "\\item 'KANSL1'\n",
       "\\item 'STK39'\n",
       "\\item 'MAPT-AS1'\n",
       "\\item 'HIP1R'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SNCA'\n",
       "2. 'LRRK2'\n",
       "3. 'KRTCAP2'\n",
       "4. 'TMEM175'\n",
       "5. 'CRHR1'\n",
       "6. 'NSF'\n",
       "7. 'KANSL1'\n",
       "8. 'STK39'\n",
       "9. 'MAPT-AS1'\n",
       "10. 'HIP1R'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SNCA\"     \"LRRK2\"    \"KRTCAP2\"  \"TMEM175\"  \"CRHR1\"    \"NSF\"     \n",
       " [7] \"KANSL1\"   \"STK39\"    \"MAPT-AS1\" \"HIP1R\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('head ~/bin/data_temp/PDgenes_sorted_geneonly.txt', intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write gene list to workspace bucket to check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Executing:  gsutil -u gp2-ipdgc-hackathon cp -r ~/bin/data_temp/PDgenes_sorted_geneonly.txt gs://fc-secure-b9f9d17f-b38c-407d-85e0-7a759f13cea0\"\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shell_do(str_glue('gsutil -u {BILLING_PROJECT_ID} cp -r ~/bin/data_temp/PDgenes_sorted_geneonly.txt {WORKSPACE_BUCKET}'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Show WebGestaltR options"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'affy_Axiom_BioBank1'</li><li>'affy_Axiom_PMRA'</li><li>'affy_Axiom_tx_v1'</li><li>'affy_GenomeWideSNP_5'</li><li>'affy_GenomeWideSNP_6'</li><li>'affy_Mapping10K_Xba142'</li><li>'affy_Mapping250K_Nsp'</li><li>'affy_Mapping250K_Sty'</li><li>'affy_Mapping50K_Hind240'</li><li>'affy_Mapping50K_Xba240'</li><li>'affy_OncoScan'</li><li>'affy_RosettaMerck_Human_RSTA'</li><li>'affy_hc_g110'</li><li>'affy_hg_focus'</li><li>'affy_hg_u133_plus_2'</li><li>'affy_hg_u133a'</li><li>'affy_hg_u133a_2'</li><li>'affy_hg_u133b'</li><li>'affy_hg_u95a'</li><li>'affy_hg_u95av2'</li><li>'affy_hg_u95b'</li><li>'affy_hg_u95c'</li><li>'affy_hg_u95d'</li><li>'affy_hg_u95e'</li><li>'affy_hta_2_0'</li><li>'affy_huex_1_0_st_v2'</li><li>'affy_hugene_1_0_st_v1'</li><li>'affy_hugene_2_0_st_v1'</li><li>'affy_hugenefl'</li><li>'affy_primeview'</li><li>'affy_u133_x3p'</li><li>'agilent_cgh_44b'</li><li>'agilent_gpl6848'</li><li>'agilent_sureprint_g3_ge_8x60k'</li><li>'agilent_sureprint_g3_ge_8x60k_v2'</li><li>'agilent_wholegenome'</li><li>'agilent_wholegenome_4x44k_v1'</li><li>'agilent_wholegenome_4x44k_v2'</li><li>'codelink_codelink'</li><li>'genome'</li><li>'genome_protein-coding'</li><li>'illumina_humanht_12_v3'</li><li>'illumina_humanht_12_v4'</li><li>'illumina_humanref_8_v3'</li><li>'illumina_humanwg_6_v1'</li><li>'illumina_humanwg_6_v2'</li><li>'illumina_humanwg_6_v3'</li><li>'phalanx_onearray'</li><li>'phosphosite'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'affy\\_Axiom\\_BioBank1'\n",
       "\\item 'affy\\_Axiom\\_PMRA'\n",
       "\\item 'affy\\_Axiom\\_tx\\_v1'\n",
       "\\item 'affy\\_GenomeWideSNP\\_5'\n",
       "\\item 'affy\\_GenomeWideSNP\\_6'\n",
       "\\item 'affy\\_Mapping10K\\_Xba142'\n",
       "\\item 'affy\\_Mapping250K\\_Nsp'\n",
       "\\item 'affy\\_Mapping250K\\_Sty'\n",
       "\\item 'affy\\_Mapping50K\\_Hind240'\n",
       "\\item 'affy\\_Mapping50K\\_Xba240'\n",
       "\\item 'affy\\_OncoScan'\n",
       "\\item 'affy\\_RosettaMerck\\_Human\\_RSTA'\n",
       "\\item 'affy\\_hc\\_g110'\n",
       "\\item 'affy\\_hg\\_focus'\n",
       "\\item 'affy\\_hg\\_u133\\_plus\\_2'\n",
       "\\item 'affy\\_hg\\_u133a'\n",
       "\\item 'affy\\_hg\\_u133a\\_2'\n",
       "\\item 'affy\\_hg\\_u133b'\n",
       "\\item 'affy\\_hg\\_u95a'\n",
       "\\item 'affy\\_hg\\_u95av2'\n",
       "\\item 'affy\\_hg\\_u95b'\n",
       "\\item 'affy\\_hg\\_u95c'\n",
       "\\item 'affy\\_hg\\_u95d'\n",
       "\\item 'affy\\_hg\\_u95e'\n",
       "\\item 'affy\\_hta\\_2\\_0'\n",
       "\\item 'affy\\_huex\\_1\\_0\\_st\\_v2'\n",
       "\\item 'affy\\_hugene\\_1\\_0\\_st\\_v1'\n",
       "\\item 'affy\\_hugene\\_2\\_0\\_st\\_v1'\n",
       "\\item 'affy\\_hugenefl'\n",
       "\\item 'affy\\_primeview'\n",
       "\\item 'affy\\_u133\\_x3p'\n",
       "\\item 'agilent\\_cgh\\_44b'\n",
       "\\item 'agilent\\_gpl6848'\n",
       "\\item 'agilent\\_sureprint\\_g3\\_ge\\_8x60k'\n",
       "\\item 'agilent\\_sureprint\\_g3\\_ge\\_8x60k\\_v2'\n",
       "\\item 'agilent\\_wholegenome'\n",
       "\\item 'agilent\\_wholegenome\\_4x44k\\_v1'\n",
       "\\item 'agilent\\_wholegenome\\_4x44k\\_v2'\n",
       "\\item 'codelink\\_codelink'\n",
       "\\item 'genome'\n",
       "\\item 'genome\\_protein-coding'\n",
       "\\item 'illumina\\_humanht\\_12\\_v3'\n",
       "\\item 'illumina\\_humanht\\_12\\_v4'\n",
       "\\item 'illumina\\_humanref\\_8\\_v3'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v1'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v2'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v3'\n",
       "\\item 'phalanx\\_onearray'\n",
       "\\item 'phosphosite'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'affy_Axiom_BioBank1'\n",
       "2. 'affy_Axiom_PMRA'\n",
       "3. 'affy_Axiom_tx_v1'\n",
       "4. 'affy_GenomeWideSNP_5'\n",
       "5. 'affy_GenomeWideSNP_6'\n",
       "6. 'affy_Mapping10K_Xba142'\n",
       "7. 'affy_Mapping250K_Nsp'\n",
       "8. 'affy_Mapping250K_Sty'\n",
       "9. 'affy_Mapping50K_Hind240'\n",
       "10. 'affy_Mapping50K_Xba240'\n",
       "11. 'affy_OncoScan'\n",
       "12. 'affy_RosettaMerck_Human_RSTA'\n",
       "13. 'affy_hc_g110'\n",
       "14. 'affy_hg_focus'\n",
       "15. 'affy_hg_u133_plus_2'\n",
       "16. 'affy_hg_u133a'\n",
       "17. 'affy_hg_u133a_2'\n",
       "18. 'affy_hg_u133b'\n",
       "19. 'affy_hg_u95a'\n",
       "20. 'affy_hg_u95av2'\n",
       "21. 'affy_hg_u95b'\n",
       "22. 'affy_hg_u95c'\n",
       "23. 'affy_hg_u95d'\n",
       "24. 'affy_hg_u95e'\n",
       "25. 'affy_hta_2_0'\n",
       "26. 'affy_huex_1_0_st_v2'\n",
       "27. 'affy_hugene_1_0_st_v1'\n",
       "28. 'affy_hugene_2_0_st_v1'\n",
       "29. 'affy_hugenefl'\n",
       "30. 'affy_primeview'\n",
       "31. 'affy_u133_x3p'\n",
       "32. 'agilent_cgh_44b'\n",
       "33. 'agilent_gpl6848'\n",
       "34. 'agilent_sureprint_g3_ge_8x60k'\n",
       "35. 'agilent_sureprint_g3_ge_8x60k_v2'\n",
       "36. 'agilent_wholegenome'\n",
       "37. 'agilent_wholegenome_4x44k_v1'\n",
       "38. 'agilent_wholegenome_4x44k_v2'\n",
       "39. 'codelink_codelink'\n",
       "40. 'genome'\n",
       "41. 'genome_protein-coding'\n",
       "42. 'illumina_humanht_12_v3'\n",
       "43. 'illumina_humanht_12_v4'\n",
       "44. 'illumina_humanref_8_v3'\n",
       "45. 'illumina_humanwg_6_v1'\n",
       "46. 'illumina_humanwg_6_v2'\n",
       "47. 'illumina_humanwg_6_v3'\n",
       "48. 'phalanx_onearray'\n",
       "49. 'phosphosite'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"affy_Axiom_BioBank1\"              \"affy_Axiom_PMRA\"                 \n",
       " [3] \"affy_Axiom_tx_v1\"                 \"affy_GenomeWideSNP_5\"            \n",
       " [5] \"affy_GenomeWideSNP_6\"             \"affy_Mapping10K_Xba142\"          \n",
       " [7] \"affy_Mapping250K_Nsp\"             \"affy_Mapping250K_Sty\"            \n",
       " [9] \"affy_Mapping50K_Hind240\"          \"affy_Mapping50K_Xba240\"          \n",
       "[11] \"affy_OncoScan\"                    \"affy_RosettaMerck_Human_RSTA\"    \n",
       "[13] \"affy_hc_g110\"                     \"affy_hg_focus\"                   \n",
       "[15] \"affy_hg_u133_plus_2\"              \"affy_hg_u133a\"                   \n",
       "[17] \"affy_hg_u133a_2\"                  \"affy_hg_u133b\"                   \n",
       "[19] \"affy_hg_u95a\"                     \"affy_hg_u95av2\"                  \n",
       "[21] \"affy_hg_u95b\"                     \"affy_hg_u95c\"                    \n",
       "[23] \"affy_hg_u95d\"                     \"affy_hg_u95e\"                    \n",
       "[25] \"affy_hta_2_0\"                     \"affy_huex_1_0_st_v2\"             \n",
       "[27] \"affy_hugene_1_0_st_v1\"            \"affy_hugene_2_0_st_v1\"           \n",
       "[29] \"affy_hugenefl\"                    \"affy_primeview\"                  \n",
       "[31] \"affy_u133_x3p\"                    \"agilent_cgh_44b\"                 \n",
       "[33] \"agilent_gpl6848\"                  \"agilent_sureprint_g3_ge_8x60k\"   \n",
       "[35] \"agilent_sureprint_g3_ge_8x60k_v2\" \"agilent_wholegenome\"             \n",
       "[37] \"agilent_wholegenome_4x44k_v1\"     \"agilent_wholegenome_4x44k_v2\"    \n",
       "[39] \"codelink_codelink\"                \"genome\"                          \n",
       "[41] \"genome_protein-coding\"            \"illumina_humanht_12_v3\"          \n",
       "[43] \"illumina_humanht_12_v4\"           \"illumina_humanref_8_v3\"          \n",
       "[45] \"illumina_humanwg_6_v1\"            \"illumina_humanwg_6_v2\"           \n",
       "[47] \"illumina_humanwg_6_v3\"            \"phalanx_onearray\"                \n",
       "[49] \"phosphosite\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listReferenceSet(\n",
    "organism = \"hsapiens\",\n",
    "hostName = \"http://www.webgestalt.org/\",\n",
    "cache = NULL\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'The_Genotype-Tissue_ExpressionProjectGTEx'</li><li>'affy_Axiom_BioBank1'</li><li>'affy_Axiom_PMRA'</li><li>'affy_Axiom_tx_v1'</li><li>'affy_GenomeWideSNP_5'</li><li>'affy_GenomeWideSNP_6'</li><li>'affy_Mapping10K_Xba142'</li><li>'affy_Mapping250K_Nsp'</li><li>'affy_Mapping250K_Sty'</li><li>'affy_Mapping50K_Hind240'</li><li>'affy_Mapping50K_Xba240'</li><li>'affy_OncoScan'</li><li>'affy_RosettaMerck_Human_RSTA'</li><li>'affy_hc_g110'</li><li>'affy_hg_focus'</li><li>'affy_hg_u133_plus_2'</li><li>'affy_hg_u133a'</li><li>'affy_hg_u133a_2'</li><li>'affy_hg_u133b'</li><li>'affy_hg_u95a'</li><li>'affy_hg_u95av2'</li><li>'affy_hg_u95b'</li><li>'affy_hg_u95c'</li><li>'affy_hg_u95d'</li><li>'affy_hg_u95e'</li><li>'affy_hta_2_0'</li><li>'affy_huex_1_0_st_v2'</li><li>'affy_hugene_1_0_st_v1'</li><li>'affy_hugene_2_0_st_v1'</li><li>'affy_hugenefl'</li><li>'affy_primeview'</li><li>'affy_u133_x3p'</li><li>'agilent_cgh_44b'</li><li>'agilent_custom_SAGE_Bionetworks_GPL564'</li><li>'agilent_gpl6848'</li><li>'agilent_human_custom_GPL564'</li><li>'agilent_sureprint_g3_ge_8x60k'</li><li>'agilent_sureprint_g3_ge_8x60k_v2'</li><li>'agilent_wholegenome'</li><li>'agilent_wholegenome_4x44k_v1'</li><li>'agilent_wholegenome_4x44k_v2'</li><li>'codelink_codelink'</li><li>'dbSNP'</li><li>'embl'</li><li>'ensembl_gene_id'</li><li>'ensembl_peptide_id'</li><li>'entrezgene'</li><li>'entrezgene_protein-coding'</li><li>'genename'</li><li>'genesymbol'</li><li>'illumina_Infinium_HumanMethylation_beadchip'</li><li>'illumina_Sentrix_HumanRef-8v2_GPL2700'</li><li>'illumina_human-6v3'</li><li>'illumina_humanRef-8v2'</li><li>'illumina_human_methylation_27'</li><li>'illumina_human_methylation_450'</li><li>'illumina_humanht_12_v3'</li><li>'illumina_humanht_12_v4'</li><li>'illumina_humanref_8_v3'</li><li>'illumina_humanwg_6_v1'</li><li>'illumina_humanwg_6_v2'</li><li>'illumina_humanwg_6_v3'</li><li>'interpro'</li><li>'phalanx_onearray'</li><li>'phosphositeEnsembl'</li><li>'phosphositeRefseq'</li><li>'phosphositeSeq'</li><li>'phosphositeUniprot'</li><li>'protein_id'</li><li>'refseq_mrna'</li><li>'refseq_peptide'</li><li>'unigene'</li><li>'uniprotswissprot'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'The\\_Genotype-Tissue\\_ExpressionProjectGTEx'\n",
       "\\item 'affy\\_Axiom\\_BioBank1'\n",
       "\\item 'affy\\_Axiom\\_PMRA'\n",
       "\\item 'affy\\_Axiom\\_tx\\_v1'\n",
       "\\item 'affy\\_GenomeWideSNP\\_5'\n",
       "\\item 'affy\\_GenomeWideSNP\\_6'\n",
       "\\item 'affy\\_Mapping10K\\_Xba142'\n",
       "\\item 'affy\\_Mapping250K\\_Nsp'\n",
       "\\item 'affy\\_Mapping250K\\_Sty'\n",
       "\\item 'affy\\_Mapping50K\\_Hind240'\n",
       "\\item 'affy\\_Mapping50K\\_Xba240'\n",
       "\\item 'affy\\_OncoScan'\n",
       "\\item 'affy\\_RosettaMerck\\_Human\\_RSTA'\n",
       "\\item 'affy\\_hc\\_g110'\n",
       "\\item 'affy\\_hg\\_focus'\n",
       "\\item 'affy\\_hg\\_u133\\_plus\\_2'\n",
       "\\item 'affy\\_hg\\_u133a'\n",
       "\\item 'affy\\_hg\\_u133a\\_2'\n",
       "\\item 'affy\\_hg\\_u133b'\n",
       "\\item 'affy\\_hg\\_u95a'\n",
       "\\item 'affy\\_hg\\_u95av2'\n",
       "\\item 'affy\\_hg\\_u95b'\n",
       "\\item 'affy\\_hg\\_u95c'\n",
       "\\item 'affy\\_hg\\_u95d'\n",
       "\\item 'affy\\_hg\\_u95e'\n",
       "\\item 'affy\\_hta\\_2\\_0'\n",
       "\\item 'affy\\_huex\\_1\\_0\\_st\\_v2'\n",
       "\\item 'affy\\_hugene\\_1\\_0\\_st\\_v1'\n",
       "\\item 'affy\\_hugene\\_2\\_0\\_st\\_v1'\n",
       "\\item 'affy\\_hugenefl'\n",
       "\\item 'affy\\_primeview'\n",
       "\\item 'affy\\_u133\\_x3p'\n",
       "\\item 'agilent\\_cgh\\_44b'\n",
       "\\item 'agilent\\_custom\\_SAGE\\_Bionetworks\\_GPL564'\n",
       "\\item 'agilent\\_gpl6848'\n",
       "\\item 'agilent\\_human\\_custom\\_GPL564'\n",
       "\\item 'agilent\\_sureprint\\_g3\\_ge\\_8x60k'\n",
       "\\item 'agilent\\_sureprint\\_g3\\_ge\\_8x60k\\_v2'\n",
       "\\item 'agilent\\_wholegenome'\n",
       "\\item 'agilent\\_wholegenome\\_4x44k\\_v1'\n",
       "\\item 'agilent\\_wholegenome\\_4x44k\\_v2'\n",
       "\\item 'codelink\\_codelink'\n",
       "\\item 'dbSNP'\n",
       "\\item 'embl'\n",
       "\\item 'ensembl\\_gene\\_id'\n",
       "\\item 'ensembl\\_peptide\\_id'\n",
       "\\item 'entrezgene'\n",
       "\\item 'entrezgene\\_protein-coding'\n",
       "\\item 'genename'\n",
       "\\item 'genesymbol'\n",
       "\\item 'illumina\\_Infinium\\_HumanMethylation\\_beadchip'\n",
       "\\item 'illumina\\_Sentrix\\_HumanRef-8v2\\_GPL2700'\n",
       "\\item 'illumina\\_human-6v3'\n",
       "\\item 'illumina\\_humanRef-8v2'\n",
       "\\item 'illumina\\_human\\_methylation\\_27'\n",
       "\\item 'illumina\\_human\\_methylation\\_450'\n",
       "\\item 'illumina\\_humanht\\_12\\_v3'\n",
       "\\item 'illumina\\_humanht\\_12\\_v4'\n",
       "\\item 'illumina\\_humanref\\_8\\_v3'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v1'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v2'\n",
       "\\item 'illumina\\_humanwg\\_6\\_v3'\n",
       "\\item 'interpro'\n",
       "\\item 'phalanx\\_onearray'\n",
       "\\item 'phosphositeEnsembl'\n",
       "\\item 'phosphositeRefseq'\n",
       "\\item 'phosphositeSeq'\n",
       "\\item 'phosphositeUniprot'\n",
       "\\item 'protein\\_id'\n",
       "\\item 'refseq\\_mrna'\n",
       "\\item 'refseq\\_peptide'\n",
       "\\item 'unigene'\n",
       "\\item 'uniprotswissprot'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'The_Genotype-Tissue_ExpressionProjectGTEx'\n",
       "2. 'affy_Axiom_BioBank1'\n",
       "3. 'affy_Axiom_PMRA'\n",
       "4. 'affy_Axiom_tx_v1'\n",
       "5. 'affy_GenomeWideSNP_5'\n",
       "6. 'affy_GenomeWideSNP_6'\n",
       "7. 'affy_Mapping10K_Xba142'\n",
       "8. 'affy_Mapping250K_Nsp'\n",
       "9. 'affy_Mapping250K_Sty'\n",
       "10. 'affy_Mapping50K_Hind240'\n",
       "11. 'affy_Mapping50K_Xba240'\n",
       "12. 'affy_OncoScan'\n",
       "13. 'affy_RosettaMerck_Human_RSTA'\n",
       "14. 'affy_hc_g110'\n",
       "15. 'affy_hg_focus'\n",
       "16. 'affy_hg_u133_plus_2'\n",
       "17. 'affy_hg_u133a'\n",
       "18. 'affy_hg_u133a_2'\n",
       "19. 'affy_hg_u133b'\n",
       "20. 'affy_hg_u95a'\n",
       "21. 'affy_hg_u95av2'\n",
       "22. 'affy_hg_u95b'\n",
       "23. 'affy_hg_u95c'\n",
       "24. 'affy_hg_u95d'\n",
       "25. 'affy_hg_u95e'\n",
       "26. 'affy_hta_2_0'\n",
       "27. 'affy_huex_1_0_st_v2'\n",
       "28. 'affy_hugene_1_0_st_v1'\n",
       "29. 'affy_hugene_2_0_st_v1'\n",
       "30. 'affy_hugenefl'\n",
       "31. 'affy_primeview'\n",
       "32. 'affy_u133_x3p'\n",
       "33. 'agilent_cgh_44b'\n",
       "34. 'agilent_custom_SAGE_Bionetworks_GPL564'\n",
       "35. 'agilent_gpl6848'\n",
       "36. 'agilent_human_custom_GPL564'\n",
       "37. 'agilent_sureprint_g3_ge_8x60k'\n",
       "38. 'agilent_sureprint_g3_ge_8x60k_v2'\n",
       "39. 'agilent_wholegenome'\n",
       "40. 'agilent_wholegenome_4x44k_v1'\n",
       "41. 'agilent_wholegenome_4x44k_v2'\n",
       "42. 'codelink_codelink'\n",
       "43. 'dbSNP'\n",
       "44. 'embl'\n",
       "45. 'ensembl_gene_id'\n",
       "46. 'ensembl_peptide_id'\n",
       "47. 'entrezgene'\n",
       "48. 'entrezgene_protein-coding'\n",
       "49. 'genename'\n",
       "50. 'genesymbol'\n",
       "51. 'illumina_Infinium_HumanMethylation_beadchip'\n",
       "52. 'illumina_Sentrix_HumanRef-8v2_GPL2700'\n",
       "53. 'illumina_human-6v3'\n",
       "54. 'illumina_humanRef-8v2'\n",
       "55. 'illumina_human_methylation_27'\n",
       "56. 'illumina_human_methylation_450'\n",
       "57. 'illumina_humanht_12_v3'\n",
       "58. 'illumina_humanht_12_v4'\n",
       "59. 'illumina_humanref_8_v3'\n",
       "60. 'illumina_humanwg_6_v1'\n",
       "61. 'illumina_humanwg_6_v2'\n",
       "62. 'illumina_humanwg_6_v3'\n",
       "63. 'interpro'\n",
       "64. 'phalanx_onearray'\n",
       "65. 'phosphositeEnsembl'\n",
       "66. 'phosphositeRefseq'\n",
       "67. 'phosphositeSeq'\n",
       "68. 'phosphositeUniprot'\n",
       "69. 'protein_id'\n",
       "70. 'refseq_mrna'\n",
       "71. 'refseq_peptide'\n",
       "72. 'unigene'\n",
       "73. 'uniprotswissprot'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"The_Genotype-Tissue_ExpressionProjectGTEx\"  \n",
       " [2] \"affy_Axiom_BioBank1\"                        \n",
       " [3] \"affy_Axiom_PMRA\"                            \n",
       " [4] \"affy_Axiom_tx_v1\"                           \n",
       " [5] \"affy_GenomeWideSNP_5\"                       \n",
       " [6] \"affy_GenomeWideSNP_6\"                       \n",
       " [7] \"affy_Mapping10K_Xba142\"                     \n",
       " [8] \"affy_Mapping250K_Nsp\"                       \n",
       " [9] \"affy_Mapping250K_Sty\"                       \n",
       "[10] \"affy_Mapping50K_Hind240\"                    \n",
       "[11] \"affy_Mapping50K_Xba240\"                     \n",
       "[12] \"affy_OncoScan\"                              \n",
       "[13] \"affy_RosettaMerck_Human_RSTA\"               \n",
       "[14] \"affy_hc_g110\"                               \n",
       "[15] \"affy_hg_focus\"                              \n",
       "[16] \"affy_hg_u133_plus_2\"                        \n",
       "[17] \"affy_hg_u133a\"                              \n",
       "[18] \"affy_hg_u133a_2\"                            \n",
       "[19] \"affy_hg_u133b\"                              \n",
       "[20] \"affy_hg_u95a\"                               \n",
       "[21] \"affy_hg_u95av2\"                             \n",
       "[22] \"affy_hg_u95b\"                               \n",
       "[23] \"affy_hg_u95c\"                               \n",
       "[24] \"affy_hg_u95d\"                               \n",
       "[25] \"affy_hg_u95e\"                               \n",
       "[26] \"affy_hta_2_0\"                               \n",
       "[27] \"affy_huex_1_0_st_v2\"                        \n",
       "[28] \"affy_hugene_1_0_st_v1\"                      \n",
       "[29] \"affy_hugene_2_0_st_v1\"                      \n",
       "[30] \"affy_hugenefl\"                              \n",
       "[31] \"affy_primeview\"                             \n",
       "[32] \"affy_u133_x3p\"                              \n",
       "[33] \"agilent_cgh_44b\"                            \n",
       "[34] \"agilent_custom_SAGE_Bionetworks_GPL564\"     \n",
       "[35] \"agilent_gpl6848\"                            \n",
       "[36] \"agilent_human_custom_GPL564\"                \n",
       "[37] \"agilent_sureprint_g3_ge_8x60k\"              \n",
       "[38] \"agilent_sureprint_g3_ge_8x60k_v2\"           \n",
       "[39] \"agilent_wholegenome\"                        \n",
       "[40] \"agilent_wholegenome_4x44k_v1\"               \n",
       "[41] \"agilent_wholegenome_4x44k_v2\"               \n",
       "[42] \"codelink_codelink\"                          \n",
       "[43] \"dbSNP\"                                      \n",
       "[44] \"embl\"                                       \n",
       "[45] \"ensembl_gene_id\"                            \n",
       "[46] \"ensembl_peptide_id\"                         \n",
       "[47] \"entrezgene\"                                 \n",
       "[48] \"entrezgene_protein-coding\"                  \n",
       "[49] \"genename\"                                   \n",
       "[50] \"genesymbol\"                                 \n",
       "[51] \"illumina_Infinium_HumanMethylation_beadchip\"\n",
       "[52] \"illumina_Sentrix_HumanRef-8v2_GPL2700\"      \n",
       "[53] \"illumina_human-6v3\"                         \n",
       "[54] \"illumina_humanRef-8v2\"                      \n",
       "[55] \"illumina_human_methylation_27\"              \n",
       "[56] \"illumina_human_methylation_450\"             \n",
       "[57] \"illumina_humanht_12_v3\"                     \n",
       "[58] \"illumina_humanht_12_v4\"                     \n",
       "[59] \"illumina_humanref_8_v3\"                     \n",
       "[60] \"illumina_humanwg_6_v1\"                      \n",
       "[61] \"illumina_humanwg_6_v2\"                      \n",
       "[62] \"illumina_humanwg_6_v3\"                      \n",
       "[63] \"interpro\"                                   \n",
       "[64] \"phalanx_onearray\"                           \n",
       "[65] \"phosphositeEnsembl\"                         \n",
       "[66] \"phosphositeRefseq\"                          \n",
       "[67] \"phosphositeSeq\"                             \n",
       "[68] \"phosphositeUniprot\"                         \n",
       "[69] \"protein_id\"                                 \n",
       "[70] \"refseq_mrna\"                                \n",
       "[71] \"refseq_peptide\"                             \n",
       "[72] \"unigene\"                                    \n",
       "[73] \"uniprotswissprot\"                           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listIdType(organism=\"hsapiens\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 64 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>name</th><th scope=col>description</th><th scope=col>idType</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>geneontology_Biological_Process            </td><td>The gene ontology biological process database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>geneontology_Biological_Process_noRedundant</td><td>The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>geneontology_Cellular_Component            </td><td>The gene ontology cellular component database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>geneontology_Cellular_Component_noRedundant</td><td>The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>geneontology_Molecular_Function            </td><td>The gene ontology molecular function database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>geneontology_Molecular_Function_noRedundant</td><td>The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>7</th><td>pathway_KEGG                               </td><td>The KEGG pathway database was downloaded from http://www.kegg.jp/.                                                                                                                                                                                                                                                                                                                                                                                                                                                        </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>8</th><td>pathway_Panther                            </td><td>The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/.                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>9</th><td>pathway_Reactome                           </td><td>The Reactome pathway database was downloaded from http://www.reactome.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                               </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>10</th><td>pathway_Wikipathway                        </td><td>The Wikipathway database was downloaded from http://www.wikipathway.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                 </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>11</th><td>pathway_Wikipathway_cancer                 </td><td>The Wikipathway database was downloaded from http://www.wikipathway.org/. This is the subset relevant to Clinical Proteomic Tumor Analysis Consortium.                                                                                                                                                                                                                                                                                                                                                                    </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>12</th><td>network_CORUM                              </td><td>The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#).                                                                                                                                                                                                                                                                                                                                                                                                               </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>13</th><td>network_CPTAC_Proteomics_BRCA              </td><td>The BRCA (Breast invasive carcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.      </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>14</th><td>network_CPTAC_Proteomics_COADREAD          </td><td>The COADREAD (Colorectal adenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.  </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>15</th><td>network_CPTAC_Proteomics_OV                </td><td>The OV (Ovarian serous cystadenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.</td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>16</th><td>network_Kinase_phosphosite                 </td><td>The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag.                                                                                                                                                                                         </td><td>phosphositeSeq</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>network_Kinase_target                      </td><td>The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network.                                                                                                                                                                                                                                                                                                                                  </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>18</th><td>network_PPI_BIOGRID                        </td><td>The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                                                                                                                                                                                                              </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>19</th><td>network_PTMsigDB                           </td><td>The database was downloaded from the PTM signatures database (https://github.com/broadinstitute/ssGSEA2.0). Direction of change is removed.                                                                                                                                                                                                                                                                                                                                                                               </td><td>phosphositeSeq</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>network_TCGA_RNASeq_ACC                    </td><td>The ACC (Adrenocortical carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                  </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>21</th><td>network_TCGA_RNASeq_BLCA                   </td><td>The BLCA (Bladder Urothelial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                             </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>22</th><td>network_TCGA_RNASeq_BRCA                   </td><td>The BRCA (Breast invasive carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>23</th><td>network_TCGA_RNASeq_CESC                   </td><td>The CESC (Cervical squamous cell carcinoma and endocervical adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.         </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>24</th><td>network_TCGA_RNASeq_CHOL                   </td><td>The CHOL (Cholangiocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                       </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>25</th><td>network_TCGA_RNASeq_COADREAD               </td><td>The COADREAD (Colon and rectal cancer) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                              </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>26</th><td>network_TCGA_RNASeq_DLBC                   </td><td>The DLBC (Lymphoid Neoplasm Diffuse Large B-cell Lymphoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                          </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>27</th><td>network_TCGA_RNASeq_ESCA                   </td><td>The ESCA (Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                     </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>28</th><td>network_TCGA_RNASeq_GBM                    </td><td>The GBM (Glioblastoma multiforme) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                   </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>29</th><td>network_TCGA_RNASeq_GBMLGG                 </td><td>The GBMLGG (Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                 </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>30</th><td>network_TCGA_RNASeq_HNSC                   </td><td>The HNSC (Head and Neck squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                    </td><td>entrezgene    </td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>35</th><td>network_TCGA_RNASeq_LIHC                       </td><td>The LIHC (Liver hepatocellular carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.      </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>36</th><td>network_TCGA_RNASeq_LUAD                       </td><td>The LUAD (Lung adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                 </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>network_TCGA_RNASeq_LUSC                       </td><td>The LUSC (Lung squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.        </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>38</th><td>network_TCGA_RNASeq_MESO                       </td><td>The MESO (Mesothelioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                        </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>39</th><td>network_TCGA_RNASeq_OV                         </td><td>The OV (Ovarian serous cystadenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.     </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>network_TCGA_RNASeq_PAAD                       </td><td>The PAAD (Pancreatic adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.           </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>41</th><td>network_TCGA_RNASeq_PCPG                       </td><td>The PCPG (Pheochromocytoma and Paraganglioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.  </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>42</th><td>network_TCGA_RNASeq_PRAD                       </td><td>The PRAD (Prostate adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.             </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>43</th><td>network_TCGA_RNASeq_SARC                       </td><td>The SARC (Sarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                             </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>44</th><td>network_TCGA_RNASeq_SKCM                       </td><td>The SKCM (Skin Cutaneous Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.             </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>45</th><td>network_TCGA_RNASeq_STAD                       </td><td>The STAD (Stomach adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.              </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>46</th><td>network_TCGA_RNASeq_STES                       </td><td>The STES (Stomach and Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.    </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>47</th><td>network_TCGA_RNASeq_TGCT                       </td><td>The TGCT (Testicular Germ Cell Tumors) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.         </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>48</th><td>network_TCGA_RNASeq_THCA                       </td><td>The THCA (Thyroid carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                   </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>49</th><td>network_TCGA_RNASeq_THYM                       </td><td>The THYM (Thymoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                             </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>50</th><td>network_TCGA_RNASeq_UCEC                       </td><td>The UCEC (Uterine Corpus Endometrial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.</td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>51</th><td>network_TCGA_RNASeq_UCS                        </td><td>The UCS (Uterine Carcinosarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.               </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>52</th><td>network_TCGA_RNASeq_UVM                        </td><td>The UVM (Uveal Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                       </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>53</th><td>network_Transcription_Factor_target            </td><td>The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                           </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>54</th><td>network_miRNA_target                           </td><td>The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                                          </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>55</th><td>disease_Disgenet                               </td><td>The DisgeNET disease database was downloaded from http://www.disgenet.org.                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>56</th><td>disease_GLAD4U                                 </td><td>Disease terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual disease term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                            </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>57</th><td>disease_OMIM                                   </td><td>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>58</th><td>drug_DrugBank                                  </td><td>The DrugBank drug database was downloaded from https://www.drugbank.ca/.                                                                                                                                                                                                                                                                                                                                                                                                             </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>59</th><td>drug_GLAD4U                                    </td><td>Drug terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual drug term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                                  </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>60</th><td>phenotype_Human_Phenotype_Ontology             </td><td>The phenotype ontology information for human was downloaded from http://human-phenotype-ontology.github.io/.                                                                                                                                                                                                                                                                                                                                                                         </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>61</th><td>chromosomalLocation_CytogeneticBand            </td><td>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>62</th><td>community-contributed_5htGeneSets_Conte        </td><td>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>63</th><td>community-contributed_Hallmark50               </td><td>Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. See details at http://software.broadinstitute.org/gsea/msigdb/collections.jsp#H .                                                                                                                                                                                                                                                                   </td><td>entrezgene</td></tr>\n",
       "\t<tr><th scope=row>64</th><td>community-contributed_MuscleGeneSets_Duddy_2017</td><td>Muscle Gene Sets (v3) Duddy 2017                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td><td>entrezgene</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 64 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & name & description & idType\\\\\n",
       "  & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & geneontology\\_Biological\\_Process             & The gene ontology biological process database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            & entrezgene    \\\\\n",
       "\t2 & geneontology\\_Biological\\_Process\\_noRedundant & The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  & entrezgene    \\\\\n",
       "\t3 & geneontology\\_Cellular\\_Component             & The gene ontology cellular component database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            & entrezgene    \\\\\n",
       "\t4 & geneontology\\_Cellular\\_Component\\_noRedundant & The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  & entrezgene    \\\\\n",
       "\t5 & geneontology\\_Molecular\\_Function             & The gene ontology molecular function database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            & entrezgene    \\\\\n",
       "\t6 & geneontology\\_Molecular\\_Function\\_noRedundant & The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  & entrezgene    \\\\\n",
       "\t7 & pathway\\_KEGG                                & The KEGG pathway database was downloaded from http://www.kegg.jp/.                                                                                                                                                                                                                                                                                                                                                                                                                                                         & entrezgene    \\\\\n",
       "\t8 & pathway\\_Panther                             & The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/.                                                                                                                                                                                                                                                                                                                                                                                                                                        & entrezgene    \\\\\n",
       "\t9 & pathway\\_Reactome                            & The Reactome pathway database was downloaded from http://www.reactome.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                & entrezgene    \\\\\n",
       "\t10 & pathway\\_Wikipathway                         & The Wikipathway database was downloaded from http://www.wikipathway.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                  & entrezgene    \\\\\n",
       "\t11 & pathway\\_Wikipathway\\_cancer                  & The Wikipathway database was downloaded from http://www.wikipathway.org/. This is the subset relevant to Clinical Proteomic Tumor Analysis Consortium.                                                                                                                                                                                                                                                                                                                                                                     & entrezgene    \\\\\n",
       "\t12 & network\\_CORUM                               & The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/\\#).                                                                                                                                                                                                                                                                                                                                                                                                                & entrezgene    \\\\\n",
       "\t13 & network\\_CPTAC\\_Proteomics\\_BRCA               & The BRCA (Breast invasive carcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.       & entrezgene    \\\\\n",
       "\t14 & network\\_CPTAC\\_Proteomics\\_COADREAD           & The COADREAD (Colorectal adenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.   & entrezgene    \\\\\n",
       "\t15 & network\\_CPTAC\\_Proteomics\\_OV                 & The OV (Ovarian serous cystadenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules. & entrezgene    \\\\\n",
       "\t16 & network\\_Kinase\\_phosphosite                  & The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/\\textasciitilde{}RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag.                                                                                                                                                                                          & phosphositeSeq\\\\\n",
       "\t17 & network\\_Kinase\\_target                       & The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network.                                                                                                                                                                                                                                                                                                                                   & entrezgene    \\\\\n",
       "\t18 & network\\_PPI\\_BIOGRID                         & The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                                                                                                                                                                                                               & entrezgene    \\\\\n",
       "\t19 & network\\_PTMsigDB                            & The database was downloaded from the PTM signatures database (https://github.com/broadinstitute/ssGSEA2.0). Direction of change is removed.                                                                                                                                                                                                                                                                                                                                                                                & phosphositeSeq\\\\\n",
       "\t20 & network\\_TCGA\\_RNASeq\\_ACC                     & The ACC (Adrenocortical carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                   & entrezgene    \\\\\n",
       "\t21 & network\\_TCGA\\_RNASeq\\_BLCA                    & The BLCA (Bladder Urothelial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                              & entrezgene    \\\\\n",
       "\t22 & network\\_TCGA\\_RNASeq\\_BRCA                    & The BRCA (Breast invasive carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                 & entrezgene    \\\\\n",
       "\t23 & network\\_TCGA\\_RNASeq\\_CESC                    & The CESC (Cervical squamous cell carcinoma and endocervical adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.          & entrezgene    \\\\\n",
       "\t24 & network\\_TCGA\\_RNASeq\\_CHOL                    & The CHOL (Cholangiocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                        & entrezgene    \\\\\n",
       "\t25 & network\\_TCGA\\_RNASeq\\_COADREAD                & The COADREAD (Colon and rectal cancer) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                               & entrezgene    \\\\\n",
       "\t26 & network\\_TCGA\\_RNASeq\\_DLBC                    & The DLBC (Lymphoid Neoplasm Diffuse Large B-cell Lymphoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                           & entrezgene    \\\\\n",
       "\t27 & network\\_TCGA\\_RNASeq\\_ESCA                    & The ESCA (Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                      & entrezgene    \\\\\n",
       "\t28 & network\\_TCGA\\_RNASeq\\_GBM                     & The GBM (Glioblastoma multiforme) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                    & entrezgene    \\\\\n",
       "\t29 & network\\_TCGA\\_RNASeq\\_GBMLGG                  & The GBMLGG (Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                  & entrezgene    \\\\\n",
       "\t30 & network\\_TCGA\\_RNASeq\\_HNSC                    & The HNSC (Head and Neck squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                     & entrezgene    \\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t35 & network\\_TCGA\\_RNASeq\\_LIHC                        & The LIHC (Liver hepatocellular carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.       & entrezgene\\\\\n",
       "\t36 & network\\_TCGA\\_RNASeq\\_LUAD                        & The LUAD (Lung adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                  & entrezgene\\\\\n",
       "\t37 & network\\_TCGA\\_RNASeq\\_LUSC                        & The LUSC (Lung squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.         & entrezgene\\\\\n",
       "\t38 & network\\_TCGA\\_RNASeq\\_MESO                        & The MESO (Mesothelioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                         & entrezgene\\\\\n",
       "\t39 & network\\_TCGA\\_RNASeq\\_OV                          & The OV (Ovarian serous cystadenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.      & entrezgene\\\\\n",
       "\t40 & network\\_TCGA\\_RNASeq\\_PAAD                        & The PAAD (Pancreatic adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.            & entrezgene\\\\\n",
       "\t41 & network\\_TCGA\\_RNASeq\\_PCPG                        & The PCPG (Pheochromocytoma and Paraganglioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.   & entrezgene\\\\\n",
       "\t42 & network\\_TCGA\\_RNASeq\\_PRAD                        & The PRAD (Prostate adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.              & entrezgene\\\\\n",
       "\t43 & network\\_TCGA\\_RNASeq\\_SARC                        & The SARC (Sarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                              & entrezgene\\\\\n",
       "\t44 & network\\_TCGA\\_RNASeq\\_SKCM                        & The SKCM (Skin Cutaneous Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.              & entrezgene\\\\\n",
       "\t45 & network\\_TCGA\\_RNASeq\\_STAD                        & The STAD (Stomach adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.               & entrezgene\\\\\n",
       "\t46 & network\\_TCGA\\_RNASeq\\_STES                        & The STES (Stomach and Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.     & entrezgene\\\\\n",
       "\t47 & network\\_TCGA\\_RNASeq\\_TGCT                        & The TGCT (Testicular Germ Cell Tumors) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.          & entrezgene\\\\\n",
       "\t48 & network\\_TCGA\\_RNASeq\\_THCA                        & The THCA (Thyroid carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                    & entrezgene\\\\\n",
       "\t49 & network\\_TCGA\\_RNASeq\\_THYM                        & The THYM (Thymoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                              & entrezgene\\\\\n",
       "\t50 & network\\_TCGA\\_RNASeq\\_UCEC                        & The UCEC (Uterine Corpus Endometrial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules. & entrezgene\\\\\n",
       "\t51 & network\\_TCGA\\_RNASeq\\_UCS                         & The UCS (Uterine Carcinosarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                & entrezgene\\\\\n",
       "\t52 & network\\_TCGA\\_RNASeq\\_UVM                         & The UVM (Uveal Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                        & entrezgene\\\\\n",
       "\t53 & network\\_Transcription\\_Factor\\_target             & The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                            & entrezgene\\\\\n",
       "\t54 & network\\_miRNA\\_target                            & The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                                           & entrezgene\\\\\n",
       "\t55 & disease\\_Disgenet                                & The DisgeNET disease database was downloaded from http://www.disgenet.org.                                                                                                                                                                                                                                                                                                                                                                                                            & entrezgene\\\\\n",
       "\t56 & disease\\_GLAD4U                                  & Disease terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual disease term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                             & entrezgene\\\\\n",
       "\t57 & disease\\_OMIM                                    &                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       & entrezgene\\\\\n",
       "\t58 & drug\\_DrugBank                                   & The DrugBank drug database was downloaded from https://www.drugbank.ca/.                                                                                                                                                                                                                                                                                                                                                                                                              & entrezgene\\\\\n",
       "\t59 & drug\\_GLAD4U                                     & Drug terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual drug term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                                   & entrezgene\\\\\n",
       "\t60 & phenotype\\_Human\\_Phenotype\\_Ontology              & The phenotype ontology information for human was downloaded from http://human-phenotype-ontology.github.io/.                                                                                                                                                                                                                                                                                                                                                                          & entrezgene\\\\\n",
       "\t61 & chromosomalLocation\\_CytogeneticBand             &                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       & entrezgene\\\\\n",
       "\t62 & community-contributed\\_5htGeneSets\\_Conte         &                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       & entrezgene\\\\\n",
       "\t63 & community-contributed\\_Hallmark50                & Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. See details at http://software.broadinstitute.org/gsea/msigdb/collections.jsp\\#H .                                                                                                                                                                                                                                                                    & entrezgene\\\\\n",
       "\t64 & community-contributed\\_MuscleGeneSets\\_Duddy\\_2017 & Muscle Gene Sets (v3) Duddy 2017                                                                                                                                                                                                                                                                                                                                                                                                                                                      & entrezgene\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 64 × 3\n",
       "\n",
       "| <!--/--> | name &lt;chr&gt; | description &lt;chr&gt; | idType &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | geneontology_Biological_Process             | The gene ontology biological process database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            | entrezgene     |\n",
       "| 2 | geneontology_Biological_Process_noRedundant | The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  | entrezgene     |\n",
       "| 3 | geneontology_Cellular_Component             | The gene ontology cellular component database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            | entrezgene     |\n",
       "| 4 | geneontology_Cellular_Component_noRedundant | The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  | entrezgene     |\n",
       "| 5 | geneontology_Molecular_Function             | The gene ontology molecular function database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                            | entrezgene     |\n",
       "| 6 | geneontology_Molecular_Function_noRedundant | The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                  | entrezgene     |\n",
       "| 7 | pathway_KEGG                                | The KEGG pathway database was downloaded from http://www.kegg.jp/.                                                                                                                                                                                                                                                                                                                                                                                                                                                         | entrezgene     |\n",
       "| 8 | pathway_Panther                             | The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/.                                                                                                                                                                                                                                                                                                                                                                                                                                        | entrezgene     |\n",
       "| 9 | pathway_Reactome                            | The Reactome pathway database was downloaded from http://www.reactome.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                | entrezgene     |\n",
       "| 10 | pathway_Wikipathway                         | The Wikipathway database was downloaded from http://www.wikipathway.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                  | entrezgene     |\n",
       "| 11 | pathway_Wikipathway_cancer                  | The Wikipathway database was downloaded from http://www.wikipathway.org/. This is the subset relevant to Clinical Proteomic Tumor Analysis Consortium.                                                                                                                                                                                                                                                                                                                                                                     | entrezgene     |\n",
       "| 12 | network_CORUM                               | The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#).                                                                                                                                                                                                                                                                                                                                                                                                                | entrezgene     |\n",
       "| 13 | network_CPTAC_Proteomics_BRCA               | The BRCA (Breast invasive carcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.       | entrezgene     |\n",
       "| 14 | network_CPTAC_Proteomics_COADREAD           | The COADREAD (Colorectal adenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.   | entrezgene     |\n",
       "| 15 | network_CPTAC_Proteomics_OV                 | The OV (Ovarian serous cystadenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules. | entrezgene     |\n",
       "| 16 | network_Kinase_phosphosite                  | The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag.                                                                                                                                                                                          | phosphositeSeq |\n",
       "| 17 | network_Kinase_target                       | The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network.                                                                                                                                                                                                                                                                                                                                   | entrezgene     |\n",
       "| 18 | network_PPI_BIOGRID                         | The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                                                                                                                                                                                                               | entrezgene     |\n",
       "| 19 | network_PTMsigDB                            | The database was downloaded from the PTM signatures database (https://github.com/broadinstitute/ssGSEA2.0). Direction of change is removed.                                                                                                                                                                                                                                                                                                                                                                                | phosphositeSeq |\n",
       "| 20 | network_TCGA_RNASeq_ACC                     | The ACC (Adrenocortical carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                   | entrezgene     |\n",
       "| 21 | network_TCGA_RNASeq_BLCA                    | The BLCA (Bladder Urothelial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                              | entrezgene     |\n",
       "| 22 | network_TCGA_RNASeq_BRCA                    | The BRCA (Breast invasive carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                 | entrezgene     |\n",
       "| 23 | network_TCGA_RNASeq_CESC                    | The CESC (Cervical squamous cell carcinoma and endocervical adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.          | entrezgene     |\n",
       "| 24 | network_TCGA_RNASeq_CHOL                    | The CHOL (Cholangiocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                        | entrezgene     |\n",
       "| 25 | network_TCGA_RNASeq_COADREAD                | The COADREAD (Colon and rectal cancer) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                               | entrezgene     |\n",
       "| 26 | network_TCGA_RNASeq_DLBC                    | The DLBC (Lymphoid Neoplasm Diffuse Large B-cell Lymphoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                           | entrezgene     |\n",
       "| 27 | network_TCGA_RNASeq_ESCA                    | The ESCA (Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                      | entrezgene     |\n",
       "| 28 | network_TCGA_RNASeq_GBM                     | The GBM (Glioblastoma multiforme) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                    | entrezgene     |\n",
       "| 29 | network_TCGA_RNASeq_GBMLGG                  | The GBMLGG (Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                  | entrezgene     |\n",
       "| 30 | network_TCGA_RNASeq_HNSC                    | The HNSC (Head and Neck squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                     | entrezgene     |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 35 | network_TCGA_RNASeq_LIHC                        | The LIHC (Liver hepatocellular carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.       | entrezgene |\n",
       "| 36 | network_TCGA_RNASeq_LUAD                        | The LUAD (Lung adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                  | entrezgene |\n",
       "| 37 | network_TCGA_RNASeq_LUSC                        | The LUSC (Lung squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.         | entrezgene |\n",
       "| 38 | network_TCGA_RNASeq_MESO                        | The MESO (Mesothelioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                         | entrezgene |\n",
       "| 39 | network_TCGA_RNASeq_OV                          | The OV (Ovarian serous cystadenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.      | entrezgene |\n",
       "| 40 | network_TCGA_RNASeq_PAAD                        | The PAAD (Pancreatic adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.            | entrezgene |\n",
       "| 41 | network_TCGA_RNASeq_PCPG                        | The PCPG (Pheochromocytoma and Paraganglioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.   | entrezgene |\n",
       "| 42 | network_TCGA_RNASeq_PRAD                        | The PRAD (Prostate adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.              | entrezgene |\n",
       "| 43 | network_TCGA_RNASeq_SARC                        | The SARC (Sarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                              | entrezgene |\n",
       "| 44 | network_TCGA_RNASeq_SKCM                        | The SKCM (Skin Cutaneous Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.              | entrezgene |\n",
       "| 45 | network_TCGA_RNASeq_STAD                        | The STAD (Stomach adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.               | entrezgene |\n",
       "| 46 | network_TCGA_RNASeq_STES                        | The STES (Stomach and Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.     | entrezgene |\n",
       "| 47 | network_TCGA_RNASeq_TGCT                        | The TGCT (Testicular Germ Cell Tumors) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.          | entrezgene |\n",
       "| 48 | network_TCGA_RNASeq_THCA                        | The THCA (Thyroid carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                    | entrezgene |\n",
       "| 49 | network_TCGA_RNASeq_THYM                        | The THYM (Thymoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                              | entrezgene |\n",
       "| 50 | network_TCGA_RNASeq_UCEC                        | The UCEC (Uterine Corpus Endometrial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules. | entrezgene |\n",
       "| 51 | network_TCGA_RNASeq_UCS                         | The UCS (Uterine Carcinosarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                | entrezgene |\n",
       "| 52 | network_TCGA_RNASeq_UVM                         | The UVM (Uveal Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                        | entrezgene |\n",
       "| 53 | network_Transcription_Factor_target             | The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                            | entrezgene |\n",
       "| 54 | network_miRNA_target                            | The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                                           | entrezgene |\n",
       "| 55 | disease_Disgenet                                | The DisgeNET disease database was downloaded from http://www.disgenet.org.                                                                                                                                                                                                                                                                                                                                                                                                            | entrezgene |\n",
       "| 56 | disease_GLAD4U                                  | Disease terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual disease term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                             | entrezgene |\n",
       "| 57 | disease_OMIM                                    | <!----> | entrezgene |\n",
       "| 58 | drug_DrugBank                                   | The DrugBank drug database was downloaded from https://www.drugbank.ca/.                                                                                                                                                                                                                                                                                                                                                                                                              | entrezgene |\n",
       "| 59 | drug_GLAD4U                                     | Drug terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual drug term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                                   | entrezgene |\n",
       "| 60 | phenotype_Human_Phenotype_Ontology              | The phenotype ontology information for human was downloaded from http://human-phenotype-ontology.github.io/.                                                                                                                                                                                                                                                                                                                                                                          | entrezgene |\n",
       "| 61 | chromosomalLocation_CytogeneticBand             | <!----> | entrezgene |\n",
       "| 62 | community-contributed_5htGeneSets_Conte         | <!----> | entrezgene |\n",
       "| 63 | community-contributed_Hallmark50                | Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. See details at http://software.broadinstitute.org/gsea/msigdb/collections.jsp#H .                                                                                                                                                                                                                                                                    | entrezgene |\n",
       "| 64 | community-contributed_MuscleGeneSets_Duddy_2017 | Muscle Gene Sets (v3) Duddy 2017                                                                                                                                                                                                                                                                                                                                                                                                                                                      | entrezgene |\n",
       "\n"
      ],
      "text/plain": [
       "   name                                           \n",
       "1  geneontology_Biological_Process                \n",
       "2  geneontology_Biological_Process_noRedundant    \n",
       "3  geneontology_Cellular_Component                \n",
       "4  geneontology_Cellular_Component_noRedundant    \n",
       "5  geneontology_Molecular_Function                \n",
       "6  geneontology_Molecular_Function_noRedundant    \n",
       "7  pathway_KEGG                                   \n",
       "8  pathway_Panther                                \n",
       "9  pathway_Reactome                               \n",
       "10 pathway_Wikipathway                            \n",
       "11 pathway_Wikipathway_cancer                     \n",
       "12 network_CORUM                                  \n",
       "13 network_CPTAC_Proteomics_BRCA                  \n",
       "14 network_CPTAC_Proteomics_COADREAD              \n",
       "15 network_CPTAC_Proteomics_OV                    \n",
       "16 network_Kinase_phosphosite                     \n",
       "17 network_Kinase_target                          \n",
       "18 network_PPI_BIOGRID                            \n",
       "19 network_PTMsigDB                               \n",
       "20 network_TCGA_RNASeq_ACC                        \n",
       "21 network_TCGA_RNASeq_BLCA                       \n",
       "22 network_TCGA_RNASeq_BRCA                       \n",
       "23 network_TCGA_RNASeq_CESC                       \n",
       "24 network_TCGA_RNASeq_CHOL                       \n",
       "25 network_TCGA_RNASeq_COADREAD                   \n",
       "26 network_TCGA_RNASeq_DLBC                       \n",
       "27 network_TCGA_RNASeq_ESCA                       \n",
       "28 network_TCGA_RNASeq_GBM                        \n",
       "29 network_TCGA_RNASeq_GBMLGG                     \n",
       "30 network_TCGA_RNASeq_HNSC                       \n",
       "⋮  ⋮                                              \n",
       "35 network_TCGA_RNASeq_LIHC                       \n",
       "36 network_TCGA_RNASeq_LUAD                       \n",
       "37 network_TCGA_RNASeq_LUSC                       \n",
       "38 network_TCGA_RNASeq_MESO                       \n",
       "39 network_TCGA_RNASeq_OV                         \n",
       "40 network_TCGA_RNASeq_PAAD                       \n",
       "41 network_TCGA_RNASeq_PCPG                       \n",
       "42 network_TCGA_RNASeq_PRAD                       \n",
       "43 network_TCGA_RNASeq_SARC                       \n",
       "44 network_TCGA_RNASeq_SKCM                       \n",
       "45 network_TCGA_RNASeq_STAD                       \n",
       "46 network_TCGA_RNASeq_STES                       \n",
       "47 network_TCGA_RNASeq_TGCT                       \n",
       "48 network_TCGA_RNASeq_THCA                       \n",
       "49 network_TCGA_RNASeq_THYM                       \n",
       "50 network_TCGA_RNASeq_UCEC                       \n",
       "51 network_TCGA_RNASeq_UCS                        \n",
       "52 network_TCGA_RNASeq_UVM                        \n",
       "53 network_Transcription_Factor_target            \n",
       "54 network_miRNA_target                           \n",
       "55 disease_Disgenet                               \n",
       "56 disease_GLAD4U                                 \n",
       "57 disease_OMIM                                   \n",
       "58 drug_DrugBank                                  \n",
       "59 drug_GLAD4U                                    \n",
       "60 phenotype_Human_Phenotype_Ontology             \n",
       "61 chromosomalLocation_CytogeneticBand            \n",
       "62 community-contributed_5htGeneSets_Conte        \n",
       "63 community-contributed_Hallmark50               \n",
       "64 community-contributed_MuscleGeneSets_Duddy_2017\n",
       "   description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "1  The gene ontology biological process database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "2  The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 \n",
       "3  The gene ontology cellular component database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "4  The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 \n",
       "5  The gene ontology molecular function database was downloaded from http://www.geneontology.org/.                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "6  The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.                                                                                                                                                                                                                 \n",
       "7  The KEGG pathway database was downloaded from http://www.kegg.jp/.                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "8  The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/.                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "9  The Reactome pathway database was downloaded from http://www.reactome.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "10 The Wikipathway database was downloaded from http://www.wikipathway.org/.                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "11 The Wikipathway database was downloaded from http://www.wikipathway.org/. This is the subset relevant to Clinical Proteomic Tumor Analysis Consortium.                                                                                                                                                                                                                                                                                                                                                                    \n",
       "12 The core complex subunits were downloaded from CORUM database (https://mips.helmholtz-muenchen.de/corum/#).                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "13 The BRCA (Breast invasive carcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.      \n",
       "14 The COADREAD (Colorectal adenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.  \n",
       "15 The OV (Ovarian serous cystadenocarcinoma) Proteomics data was downloaded from CPTAC (Clinical Proteomic Tumor Analysis Consortium) data portal (https://cptac-data-portal.georgetown.edu/cptacPublic/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus Proteomics co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.\n",
       "16 The relationship between Kinase and the corresponding protein phosposite position was from RegPhos (http://140.138.144.141/~RegPhos/). Then, we got the 15-mer sequence window around the phosphorylated site of the peptide (up and down 7 amino acid) from the Uniprot website (http://www.uniprot.org/) as the identifier tag.                                                                                                                                                                                         \n",
       "17 The relationship between Kinase and its targets was from PhosphositePlus (http://www.phosphosite.org/) and the phosphorylation relationships from a in-house combined signaling network.                                                                                                                                                                                                                                                                                                                                  \n",
       "18 The protein-protein interaction (PPI) network was downloaded from BIOGRID (https://thebiogrid.org/). Then, we used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                                                                                                                                                                                                              \n",
       "19 The database was downloaded from the PTM signatures database (https://github.com/broadinstitute/ssGSEA2.0). Direction of change is removed.                                                                                                                                                                                                                                                                                                                                                                               \n",
       "20 The ACC (Adrenocortical carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                  \n",
       "21 The BLCA (Bladder Urothelial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                             \n",
       "22 The BRCA (Breast invasive carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                \n",
       "23 The CESC (Cervical squamous cell carcinoma and endocervical adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.         \n",
       "24 The CHOL (Cholangiocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                       \n",
       "25 The COADREAD (Colon and rectal cancer) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                              \n",
       "26 The DLBC (Lymphoid Neoplasm Diffuse Large B-cell Lymphoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                          \n",
       "27 The ESCA (Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                     \n",
       "28 The GBM (Glioblastoma multiforme) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                   \n",
       "29 The GBMLGG (Glioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                 \n",
       "30 The HNSC (Head and Neck squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                    \n",
       "⋮  ⋮                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "35 The LIHC (Liver hepatocellular carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                           \n",
       "36 The LUAD (Lung adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                      \n",
       "37 The LUSC (Lung squamous cell carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                             \n",
       "38 The MESO (Mesothelioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                             \n",
       "39 The OV (Ovarian serous cystadenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                          \n",
       "40 The PAAD (Pancreatic adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                \n",
       "41 The PCPG (Pheochromocytoma and Paraganglioma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                       \n",
       "42 The PRAD (Prostate adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                  \n",
       "43 The SARC (Sarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                  \n",
       "44 The SKCM (Skin Cutaneous Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                  \n",
       "45 The STAD (Stomach adenocarcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                   \n",
       "46 The STES (Stomach and Esophageal carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                         \n",
       "47 The TGCT (Testicular Germ Cell Tumors) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                              \n",
       "48 The THCA (Thyroid carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                        \n",
       "49 The THYM (Thymoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                                  \n",
       "50 The UCEC (Uterine Corpus Endometrial Carcinoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                     \n",
       "51 The UCS (Uterine Carcinosarcoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                    \n",
       "52 The UVM (Uveal Melanoma) RNASeq data was downloaded from TCGA (The Cancer Genome Atlas) firehose data portal (https://gdac.broadinstitute.org/). Then, based on the method described in our recently published paper (PMID: 27836980), we first constructed the consensus RNASeq co-expression network and then used the NetSAM R package (http://bioconductor.org/packages/release/bioc/html/NetSAM.html) to identify the hierarchical co-expression modules.                                                            \n",
       "53 The relationship between transcription factor and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                                                                \n",
       "54 The relationship between miRNA and its targets was downloaded from MsigDB (http://software.broadinstitute.org/gsea/msigdb).                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "55 The DisgeNET disease database was downloaded from http://www.disgenet.org.                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "56 Disease terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual disease term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                                                                 \n",
       "57                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "58 The DrugBank drug database was downloaded from https://www.drugbank.ca/.                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "59 Drug terms were downloaded from ParmGKB (https://www.pharmgkb.org/). Genes associated with individual drug term were inferred using GLAD4U (http://glad4u.zhang-lab.org/index.php).                                                                                                                                                                                                                                                                                                                                       \n",
       "60 The phenotype ontology information for human was downloaded from http://human-phenotype-ontology.github.io/.                                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "61                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "62                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "63 Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. See details at http://software.broadinstitute.org/gsea/msigdb/collections.jsp#H .                                                                                                                                                                                                                                                                                                        \n",
       "64 Muscle Gene Sets (v3) Duddy 2017                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "   idType        \n",
       "1  entrezgene    \n",
       "2  entrezgene    \n",
       "3  entrezgene    \n",
       "4  entrezgene    \n",
       "5  entrezgene    \n",
       "6  entrezgene    \n",
       "7  entrezgene    \n",
       "8  entrezgene    \n",
       "9  entrezgene    \n",
       "10 entrezgene    \n",
       "11 entrezgene    \n",
       "12 entrezgene    \n",
       "13 entrezgene    \n",
       "14 entrezgene    \n",
       "15 entrezgene    \n",
       "16 phosphositeSeq\n",
       "17 entrezgene    \n",
       "18 entrezgene    \n",
       "19 phosphositeSeq\n",
       "20 entrezgene    \n",
       "21 entrezgene    \n",
       "22 entrezgene    \n",
       "23 entrezgene    \n",
       "24 entrezgene    \n",
       "25 entrezgene    \n",
       "26 entrezgene    \n",
       "27 entrezgene    \n",
       "28 entrezgene    \n",
       "29 entrezgene    \n",
       "30 entrezgene    \n",
       "⋮  ⋮             \n",
       "35 entrezgene    \n",
       "36 entrezgene    \n",
       "37 entrezgene    \n",
       "38 entrezgene    \n",
       "39 entrezgene    \n",
       "40 entrezgene    \n",
       "41 entrezgene    \n",
       "42 entrezgene    \n",
       "43 entrezgene    \n",
       "44 entrezgene    \n",
       "45 entrezgene    \n",
       "46 entrezgene    \n",
       "47 entrezgene    \n",
       "48 entrezgene    \n",
       "49 entrezgene    \n",
       "50 entrezgene    \n",
       "51 entrezgene    \n",
       "52 entrezgene    \n",
       "53 entrezgene    \n",
       "54 entrezgene    \n",
       "55 entrezgene    \n",
       "56 entrezgene    \n",
       "57 entrezgene    \n",
       "58 entrezgene    \n",
       "59 entrezgene    \n",
       "60 entrezgene    \n",
       "61 entrezgene    \n",
       "62 entrezgene    \n",
       "63 entrezgene    \n",
       "64 entrezgene    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listGeneSet(organism = \"hsapiens\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run WebGestaltR Overrepresentation Analysis (ORA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run with gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Summarizing the input ID list by GO Slim data...\n",
      "Performing the enrichment analysis...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum = minNum, :\n",
      "“No significant gene set is identified based on FDR 0.05!”\n"
     ]
    }
   ],
   "source": [
    "refFile <- system.file(\"extdata\", \"referenceGenes.txt\", package=\"WebGestaltR\")\n",
    "enrichResult <- WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",\n",
    "                            enrichDatabase=\"pathway_KEGG\", interestGeneFile=\"~/bin/data_temp/PDgenes_sorted_geneonly.txt\",\n",
    "                            interestGeneType=\"genesymbol\", referenceGeneFile=refFile,\n",
    "                            referenceGeneType=\"genesymbol\", isOutput=TRUE,\n",
    "                            outputDirectory=\"~/bin/data_temp/\", projectName=NULL)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run with SNP list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Loading the reference list...\n",
      "Summarizing the input ID list by GO Slim data...\n",
      "Performing the enrichment analysis...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum = minNum, :\n",
      "“No significant gene set is identified based on FDR 0.05!”\n"
     ]
    }
   ],
   "source": [
    "refFile <- system.file(\"extdata\", \"referenceGenes.txt\", package=\"WebGestaltR\")\n",
    "enrichResult <- WebGestaltR(enrichMethod=\"ORA\", organism=\"hsapiens\",\n",
    "                            enrichDatabase=\"pathway_KEGG\", interestGeneFile=\"~/bin/data_temp/PDSNPs_sorted.txt\",\n",
    "                            interestGeneType=\"dbSNP\", referenceGeneFile=refFile,\n",
    "                            referenceGeneType=\"genesymbol\", isOutput=TRUE,\n",
    "                            outputDirectory=\"~/bin/data_temp/\", projectName=NULL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run WebGestaltR Gene Set Enrichment Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Summarizing the uploaded ID list by GO Slim data...\n",
      "Performing the enrichment analysis...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000041\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000075\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000209\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000910\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001101\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001503\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001525\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001539\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001655\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001667\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001763\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001764\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001818\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001819\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001906\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001963\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002237\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002244\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002250\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002269\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002285\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002440\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002446\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002449\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002521\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002526\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002532\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002576\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002683\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002697\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0003012\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0003013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006091\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006109\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006260\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006289\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006302\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006308\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006310\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006338\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006352\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006353\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006360\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006397\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006401\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006413\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006457\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006520\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006575\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006631\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006638\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006643\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006720\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006766\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006790\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006814\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006839\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006909\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006925\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006959\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006968\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006970\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006979\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007006\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007015\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007018\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007033\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007050\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007059\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007162\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007163\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007178\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007187\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007212\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007215\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007219\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007229\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007271\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007272\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007274\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007369\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007389\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007498\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007517\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007548\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007586\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007626\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007631\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008202\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008213\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008277\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008380\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008544\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009100\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009123\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009132\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009141\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009266\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009268\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009308\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009314\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009410\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009581\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009582\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009593\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009595\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009615\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009743\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009755\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009914\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010171\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010498\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010644\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010657\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010737\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010821\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010876\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010948\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0014812\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015695\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015696\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015698\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015711\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015748\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015837\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015844\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015849\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015850\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015893\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016042\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016051\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016052\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016053\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016311\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016358\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016999\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0017038\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0017156\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018195\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018205\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018210\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019080\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019216\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019233\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019748\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019882\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019932\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021510\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021700\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021953\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022406\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022412\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022613\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022616\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030048\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030099\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030522\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030534\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030705\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030901\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030902\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031098\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031099\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031109\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031128\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031214\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031349\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031503\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031589\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031644\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031647\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031667\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032102\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032103\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032259\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032355\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032409\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032528\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032609\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032613\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032620\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032635\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032886\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032941\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032963\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032970\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032984\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033002\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033028\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033044\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033500\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033555\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033619\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033865\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034067\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034248\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034330\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034340\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034341\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034394\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034764\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034976\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035265\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035418\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035640\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035690\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035821\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035966\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036119\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036230\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036297\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036314\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038061\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038127\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0040013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042092\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042113\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042116\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042180\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042326\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042398\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042445\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042476\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042490\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042558\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042692\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042737\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042770\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042982\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043062\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043087\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043254\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043270\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043279\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043491\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043543\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043583\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043620\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043954\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044070\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044262\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044282\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044772\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044843\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045088\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045117\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045165\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045444\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045787\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045861\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045926\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045927\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045930\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046390\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046434\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046677\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046683\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046717\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046847\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048008\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048010\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048017\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048284\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048285\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048511\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048545\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048678\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048732\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048736\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048771\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048863\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048871\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048872\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048880\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050435\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050663\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050673\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050690\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050803\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050817\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050866\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050878\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050879\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050890\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050905\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050906\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050918\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050919\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050920\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050953\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050954\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051047\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051052\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051090\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051098\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051169\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051181\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051187\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051188\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051193\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051258\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051260\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051271\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051291\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051321\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051341\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051346\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051604\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051646\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051668\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051701\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051702\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051865\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0052547\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0055067\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0055123\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060191\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060326\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060359\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060485\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060491\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060537\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060759\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061025\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061351\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061448\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061458\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061512\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061564\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061919\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0062012\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070085\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070371\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070482\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070555\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070585\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070661\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070841\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070972\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071103\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071216\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071526\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071542\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071559\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071774\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071824\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071825\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071826\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071867\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071887\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072089\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072331\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072347\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072348\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072376\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072507\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072522\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072524\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072665\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090130\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090150\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090287\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090305\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090382\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090559\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090596\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097066\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097164\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097191\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097193\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097194\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097305\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097485\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097581\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097696\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098542\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098656\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098732\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098739\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098742\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098780\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098876\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0099072\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0104004\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0106027\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0150076\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0198738\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901136\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901293\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901342\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901652\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901657\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901987\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901998\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902115\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902532\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902579\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902600\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902903\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903008\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903311\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903320\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903509\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903828\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1904950\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1905475\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1990089\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1990868\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001020\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001057\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001233\n",
      "”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000 permutations of score complete...\n",
      "Begin affinity propagation...\n",
      "End affinity propagation...\n",
      "Begin weighted set cover...\n",
      "Remain is 0, ending weighted set cover\n",
      "Generate the final report...\n",
      "Results can be found in the ~/bin/data_temp/Project_1620831213!\n"
     ]
    }
   ],
   "source": [
    "#Run with original pvalues\n",
    "rankFile <- system.file(\"extdata\", \"GeneRankList.rnk\", package=\"WebGestaltR\")\n",
    "enrichResult <- WebGestaltR(enrichMethod=\"GSEA\", organism=\"hsapiens\",\n",
    "enrichDatabase=\"geneontology_Biological_Process_noRedundant\", interestGeneFile=\"~/bin/data_temp/PDgenes_sorted_pvals.rnk\",\n",
    "interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10, minNum=5,\n",
    "outputDirectory=\"~/bin/data_temp\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'enriched_geneset_ap_clusters_1620831213.txt'</li><li>'enriched_geneset_wsc_topsets_1620831213.txt'</li><li>'enrichment_results_1620831213.txt'</li><li>'goslim_summary_1620831213_bp.txt'</li><li>'goslim_summary_1620831213_cc.txt'</li><li>'goslim_summary_1620831213_mf.txt'</li><li>'goslim_summary_1620831213.png'</li><li>'interestingID_mappingTable_1620831213.txt'</li><li>'interestingID_unmappedList_1620831213.txt'</li><li>'Project_1620831213_GSEA'</li><li>'Project_1620831213_GSEA.rnk'</li><li>'Project_1620831213.zip'</li><li>'Report_1620831213.html'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'enriched\\_geneset\\_ap\\_clusters\\_1620831213.txt'\n",
       "\\item 'enriched\\_geneset\\_wsc\\_topsets\\_1620831213.txt'\n",
       "\\item 'enrichment\\_results\\_1620831213.txt'\n",
       "\\item 'goslim\\_summary\\_1620831213\\_bp.txt'\n",
       "\\item 'goslim\\_summary\\_1620831213\\_cc.txt'\n",
       "\\item 'goslim\\_summary\\_1620831213\\_mf.txt'\n",
       "\\item 'goslim\\_summary\\_1620831213.png'\n",
       "\\item 'interestingID\\_mappingTable\\_1620831213.txt'\n",
       "\\item 'interestingID\\_unmappedList\\_1620831213.txt'\n",
       "\\item 'Project\\_1620831213\\_GSEA'\n",
       "\\item 'Project\\_1620831213\\_GSEA.rnk'\n",
       "\\item 'Project\\_1620831213.zip'\n",
       "\\item 'Report\\_1620831213.html'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'enriched_geneset_ap_clusters_1620831213.txt'\n",
       "2. 'enriched_geneset_wsc_topsets_1620831213.txt'\n",
       "3. 'enrichment_results_1620831213.txt'\n",
       "4. 'goslim_summary_1620831213_bp.txt'\n",
       "5. 'goslim_summary_1620831213_cc.txt'\n",
       "6. 'goslim_summary_1620831213_mf.txt'\n",
       "7. 'goslim_summary_1620831213.png'\n",
       "8. 'interestingID_mappingTable_1620831213.txt'\n",
       "9. 'interestingID_unmappedList_1620831213.txt'\n",
       "10. 'Project_1620831213_GSEA'\n",
       "11. 'Project_1620831213_GSEA.rnk'\n",
       "12. 'Project_1620831213.zip'\n",
       "13. 'Report_1620831213.html'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"enriched_geneset_ap_clusters_1620831213.txt\"\n",
       " [2] \"enriched_geneset_wsc_topsets_1620831213.txt\"\n",
       " [3] \"enrichment_results_1620831213.txt\"          \n",
       " [4] \"goslim_summary_1620831213_bp.txt\"           \n",
       " [5] \"goslim_summary_1620831213_cc.txt\"           \n",
       " [6] \"goslim_summary_1620831213_mf.txt\"           \n",
       " [7] \"goslim_summary_1620831213.png\"              \n",
       " [8] \"interestingID_mappingTable_1620831213.txt\"  \n",
       " [9] \"interestingID_unmappedList_1620831213.txt\"  \n",
       "[10] \"Project_1620831213_GSEA\"                    \n",
       "[11] \"Project_1620831213_GSEA.rnk\"                \n",
       "[12] \"Project_1620831213.zip\"                     \n",
       "[13] \"Report_1620831213.html\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system('ls ~/bin/data_temp/Project_1620831213', intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write results to workspace bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Executing:  gsutil -u gp2-ipdgc-hackathon cp -r ~/bin/data_temp/Project_1620831213/goslim_summary_1620831213.png gs://fc-secure-b9f9d17f-b38c-407d-85e0-7a759f13cea0\"\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shell_do(str_glue('gsutil -u {BILLING_PROJECT_ID} cp -r ~/bin/data_temp/Project_1620831213/goslim_summary_1620831213.png {WORKSPACE_BUCKET}'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading the functional categories...\n",
      "Loading the ID list...\n",
      "Summarizing the uploaded ID list by GO Slim data...\n",
      "Performing the enrichment analysis...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000041\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000075\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000209\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0000910\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001101\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001503\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001525\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001539\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001655\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001667\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001763\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001764\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001818\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001819\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001906\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0001963\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002237\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002244\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002250\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002269\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002285\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002440\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002446\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002449\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002521\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002526\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002532\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002576\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002683\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0002697\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0003012\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0003013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006091\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006109\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006260\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006289\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006302\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006308\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006310\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006338\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006352\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006353\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006360\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006397\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006401\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006413\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006457\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006520\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006575\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006631\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006638\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006643\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006720\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006766\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006790\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006814\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006839\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006909\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006925\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006959\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006968\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006970\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0006979\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007006\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007015\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007018\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007033\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007050\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007059\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007162\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007163\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007178\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007187\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007212\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007215\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007219\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007229\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007271\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007272\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007274\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007369\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007389\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007498\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007517\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007548\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007586\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007626\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0007631\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008202\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008213\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008277\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008380\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008544\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0008637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009100\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009123\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009132\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009141\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009266\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009268\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009308\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009314\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009410\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009581\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009582\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009593\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009595\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009615\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009743\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009755\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0009914\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010171\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010498\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010644\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010657\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010737\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010821\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010876\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0010948\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0014812\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015695\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015696\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015698\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015711\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015748\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015837\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015844\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015849\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015850\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0015893\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016042\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016051\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016052\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016053\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016311\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016358\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0016999\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0017038\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0017156\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018195\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018205\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0018210\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019080\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019216\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019233\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019748\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019882\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0019932\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021510\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021700\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0021953\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022406\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022412\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022613\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0022616\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030048\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030099\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030522\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030534\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030705\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030901\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0030902\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031098\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031099\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031109\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031128\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031214\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031349\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031503\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031589\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031644\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031647\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0031667\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032102\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032103\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032259\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032355\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032409\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032528\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032609\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032613\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032620\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032635\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032886\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032941\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032963\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032970\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0032984\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033002\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033028\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033044\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033500\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033555\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033619\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0033865\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034067\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034248\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034330\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034340\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034341\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034394\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034612\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034764\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0034976\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035265\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035418\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035637\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035640\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035690\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035821\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0035966\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036119\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036230\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036297\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0036314\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038061\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0038127\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0040013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042092\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042113\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042116\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042180\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042326\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042398\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042445\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042476\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042490\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042558\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042692\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042737\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042770\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0042982\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043062\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043087\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043254\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043270\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043279\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043491\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043543\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043583\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043620\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0043954\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044070\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044262\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044282\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044772\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0044843\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045088\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045117\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045165\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045444\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045787\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045861\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045926\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045927\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0045930\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046390\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046434\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046677\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046683\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046717\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0046847\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048008\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048010\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048013\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048017\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048284\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048285\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048511\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048545\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048678\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048732\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048736\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048771\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048863\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048871\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048872\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0048880\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050435\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050663\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050673\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050690\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050803\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050817\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050866\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050878\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050879\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050890\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050900\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050905\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050906\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050918\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050919\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050920\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050953\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0050954\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051047\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051052\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051090\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051098\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051169\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051181\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051187\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051188\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051193\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051258\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051260\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051271\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051291\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051321\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051341\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051346\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051604\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051646\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051668\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051701\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051702\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0051865\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0052547\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0055067\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0055123\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060191\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060249\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060326\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060359\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060485\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060491\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060537\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0060759\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061025\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061351\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061448\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061458\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061512\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061564\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0061919\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0062012\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070085\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070371\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070482\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070555\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070585\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070661\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070841\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0070972\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071103\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071216\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071526\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071542\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071559\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071774\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071824\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071825\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071826\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071867\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0071887\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072089\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072331\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072347\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072348\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072376\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072507\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072522\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072524\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0072665\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090130\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090150\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090287\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090305\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090382\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090559\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0090596\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097066\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097164\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097191\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097193\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097194\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097305\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097485\n",
      "”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097581\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0097696\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098542\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098656\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098732\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098739\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098742\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098780\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0098876\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0099072\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0104004\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0106027\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0150076\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:0198738\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901136\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901293\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901342\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901568\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901652\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901657\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901987\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1901998\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902115\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902532\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902579\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902600\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1902903\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903008\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903034\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903311\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903320\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903509\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903706\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1903828\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1904950\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1905475\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1990089\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:1990868\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001020\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001057\n",
      "”\n",
      "Warning message in swGsea(inputDf, thresh_type = \"val\", perms = perNum, min_set_size = minNum, :\n",
      "“score does not contain minimum number of items in set for GO:2001233\n",
      "”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000 permutations of score complete...\n",
      "Begin affinity propagation...\n",
      "End affinity propagation...\n",
      "Begin weighted set cover...\n",
      "Remain is 0, ending weighted set cover\n",
      "Generate the final report...\n",
      "Results can be found in the ~/bin/data_temp/Project_1620831683!\n"
     ]
    }
   ],
   "source": [
    "#Run with -log10 pvalues\n",
    "rankFile <- system.file(\"extdata\", \"GeneRankList.rnk\", package=\"WebGestaltR\")\n",
    "enrichResult <- WebGestaltR(enrichMethod=\"GSEA\", organism=\"hsapiens\",\n",
    "enrichDatabase=\"geneontology_Biological_Process_noRedundant\", interestGeneFile=\"~/bin/data_temp/PDgenes_sorted_pvals_neglog.rnk\",\n",
    "interestGeneType=\"genesymbol\", sigMethod=\"top\", topThr=10, minNum=5,\n",
    "outputDirectory=\"~/bin/data_temp\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'enriched_geneset_ap_clusters_1620831683.txt'</li><li>'enriched_geneset_wsc_topsets_1620831683.txt'</li><li>'enrichment_results_1620831683.txt'</li><li>'goslim_summary_1620831683_bp.txt'</li><li>'goslim_summary_1620831683_cc.txt'</li><li>'goslim_summary_1620831683_mf.txt'</li><li>'goslim_summary_1620831683.png'</li><li>'interestingID_mappingTable_1620831683.txt'</li><li>'interestingID_unmappedList_1620831683.txt'</li><li>'Project_1620831683_GSEA'</li><li>'Project_1620831683_GSEA.rnk'</li><li>'Project_1620831683.zip'</li><li>'Report_1620831683.html'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'enriched\\_geneset\\_ap\\_clusters\\_1620831683.txt'\n",
       "\\item 'enriched\\_geneset\\_wsc\\_topsets\\_1620831683.txt'\n",
       "\\item 'enrichment\\_results\\_1620831683.txt'\n",
       "\\item 'goslim\\_summary\\_1620831683\\_bp.txt'\n",
       "\\item 'goslim\\_summary\\_1620831683\\_cc.txt'\n",
       "\\item 'goslim\\_summary\\_1620831683\\_mf.txt'\n",
       "\\item 'goslim\\_summary\\_1620831683.png'\n",
       "\\item 'interestingID\\_mappingTable\\_1620831683.txt'\n",
       "\\item 'interestingID\\_unmappedList\\_1620831683.txt'\n",
       "\\item 'Project\\_1620831683\\_GSEA'\n",
       "\\item 'Project\\_1620831683\\_GSEA.rnk'\n",
       "\\item 'Project\\_1620831683.zip'\n",
       "\\item 'Report\\_1620831683.html'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'enriched_geneset_ap_clusters_1620831683.txt'\n",
       "2. 'enriched_geneset_wsc_topsets_1620831683.txt'\n",
       "3. 'enrichment_results_1620831683.txt'\n",
       "4. 'goslim_summary_1620831683_bp.txt'\n",
       "5. 'goslim_summary_1620831683_cc.txt'\n",
       "6. 'goslim_summary_1620831683_mf.txt'\n",
       "7. 'goslim_summary_1620831683.png'\n",
       "8. 'interestingID_mappingTable_1620831683.txt'\n",
       "9. 'interestingID_unmappedList_1620831683.txt'\n",
       "10. 'Project_1620831683_GSEA'\n",
       "11. 'Project_1620831683_GSEA.rnk'\n",
       "12. 'Project_1620831683.zip'\n",
       "13. 'Report_1620831683.html'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"enriched_geneset_ap_clusters_1620831683.txt\"\n",
       " [2] \"enriched_geneset_wsc_topsets_1620831683.txt\"\n",
       " [3] \"enrichment_results_1620831683.txt\"          \n",
       " [4] \"goslim_summary_1620831683_bp.txt\"           \n",
       " [5] \"goslim_summary_1620831683_cc.txt\"           \n",
       " [6] \"goslim_summary_1620831683_mf.txt\"           \n",
       " [7] \"goslim_summary_1620831683.png\"              \n",
       " [8] \"interestingID_mappingTable_1620831683.txt\"  \n",
       " [9] \"interestingID_unmappedList_1620831683.txt\"  \n",
       "[10] \"Project_1620831683_GSEA\"                    \n",
       "[11] \"Project_1620831683_GSEA.rnk\"                \n",
       "[12] \"Project_1620831683.zip\"                     \n",
       "[13] \"Report_1620831683.html\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Look at the files in the results folder\n",
    "system('ls ~/bin/data_temp/Project_1620831683', intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write results to workspace bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Executing:  gsutil -u gp2-ipdgc-hackathon cp -r ~/bin/data_temp/Project_1620831683/goslim_summary_1620831683.png gs://fc-secure-b9f9d17f-b38c-407d-85e0-7a759f13cea0\"\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shell_do(str_glue('gsutil -u {BILLING_PROJECT_ID} cp -r ~/bin/data_temp/Project_1620831683/goslim_summary_1620831683.png {WORKSPACE_BUCKET}'))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
